{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a544191f-31fa-48c7-8ce2-047940ae909a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from ISResMat import TrainApp\n",
    "import argparse\n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "403e3005-c93d-4fda-827a-cc46fb483d08",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-10-11 04:48:29 INFO Using GPU.\n",
      "2024-10-11 04:48:30 INFO begin to prepare the data ...\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:79: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col] = self.df[col]\n",
      "/scratch/yfw215/data-integration-eval/data_harmonization_benchmark/matchers/ISResMat/src/data_loader/sm_dataset.py:80: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.bin_df[col + \"bin\"] = pd.cut(\n",
      "2024-10-11 04:48:32 INFO src effective table width: 287\n",
      "2024-10-11 04:48:32 INFO tgt effective table width: 734\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[125, 121, 137, 214, 209, 172, 152, 92, 79, 278, 72, 131, 147, 6, 9, 223, 208, 245, 57, 264, 249, 146, 207, 284, 38, 39, 26, 277, 43, 286, 143, 244, 161, 247, 97, 138, 153, 205, 285, 173, 44, 193, 21, 14, 107, 77, 229, 210, 269, 263, 197, 129, 45, 279, 181, 275, 199, 189, 136, 233, 155, 248, 178, 256, 130, 174, 204, 183, 109, 267, 241, 235, 60, 246, 211, 29, 41, 123, 16, 253, 230, 206, 254, 281, 81, 177, 65, 283, 203, 221, 187, 95, 171, 31, 265, 191, 86, 33, 127, 2, 103, 222, 61, 227, 67, 200, 98, 157, 80, 4, 118, 217, 112, 202, 270, 215, 11, 71, 52, 46, 149, 196, 23, 175, 88, 236, 139, 122, 163, 10, 250, 27, 28, 258, 89, 218, 226, 167, 51, 274, 216, 154, 231, 8, 165, 7, 84, 22, 120, 176, 142, 105, 5, 128, 59, 78, 198, 180, 266, 156, 282, 268, 212, 132, 53, 225, 58, 116, 134, 179, 144, 110, 19, 18, 76, 50, 150, 239, 159, 106, 91, 257, 117, 242, 158, 83, 40, 166, 169, 259, 224, 68, 85, 133, 12, 3, 141, 262, 273, 102, 37, 186, 124, 34, 243, 220, 82, 36, 24, 13, 47, 101, 237, 140, 185, 251, 213, 25, 69, 219, 100, 234, 113, 74, 188, 145, 162, 35, 148, 15, 54, 49, 42, 228, 164, 66, 1, 170, 99, 168, 190, 255, 184, 87, 119, 252, 114, 135, 30, 232, 194, 111, 70, 195, 94, 64, 32, 280, 62, 160, 115, 0, 93, 201, 271, 20, 56, 108, 261, 151, 182, 238, 73, 104, 240, 90, 48, 276, 96, 272, 126, 192, 75, 55, 260, 63, 17]\n",
      "[[6, 124, 60, 134, 111, 268], [7, 394, 318, 42, 180, 370], [391, 130, 73, 43, 72, 201], [371, 225, 182, 211, 222, 172], [229, 26, 364, 249, 92, 197], [57, 370, 41, 46, 197, 117], [150, 279, 186, 386, 293, 277], [385, 352, 302, 101, 343, 320], [102, 289, 307, 223, 230, 132], [228, 18, 281, 321, 336, 108], [61, 383, 327, 92, 50, 386], [270, 397, 363, 398, 122, 343], [343, 142, 114, 260, 150, 164], [369, 199, 29, 372, 113, 234], [328, 292, 206, 39, 232, 63], [249, 232, 305, 280, 328, 71], [150, 308, 372, 92, 233, 242], [76, 234, 264, 69, 146, 390], [76, 360, 333, 170, 271, 201], [45, 37, 61, 334, 109, 171], [298, 356, 87, 61, 334, 11], [306, 104, 77, 49, 314, 325], [208, 139, 79, 362, 168, 126], [111, 380, 329, 101, 287, 246], [65, 345, 319, 233, 27, 174], [98, 345, 314, 394, 47, 21], [180, 262, 311, 132, 347, 152], [388, 76, 57, 146, 320, 326], [165, 194, 352, 219, 28, 368], [341, 325, 393, 102, 84, 137], [240, 115, 124, 388, 381, 141], [198, 153, 256, 24, 354, 386], [213, 88, 304, 150, 333, 48], [31, 108, 110, 297, 334, 39], [123, 339, 132, 395, 347, 1], [285, 210, 163, 322, 156, 354], [322, 183, 342, 244, 269, 324], [221, 14, 0, 303, 366, 374], [120, 288, 243, 201, 321, 145], [48, 146, 52, 363, 192, 314], [196, 234, 399, 112, 314, 318], [140, 302, 365, 169, 51, 253], [7, 64, 5, 172, 18, 129], [193, 80, 379, 334, 237, 256], [20, 28, 343, 317, 346, 0], [53, 375, 339, 195, 161, 159], [263, 235, 294, 109, 194, 95], [63, 220, 180, 225, 312, 205], [69, 392, 336, 40, 326, 234], [287, 159, 38, 135, 173, 183], [115, 196, 106, 233, 42, 288], [95, 285, 360, 309, 229, 399], [196, 381, 253, 81, 297, 154], [223, 390, 200, 295, 175, 199], [284, 229, 86, 207, 253, 379], [142, 128, 82, 367, 108, 194], [87, 323, 279, 370, 390, 222], [159, 180, 227, 114, 126, 221], [172, 355, 290, 252, 248, 204], [388, 319, 37, 397, 106, 337], [86, 254, 122, 27, 224, 116], [150, 270, 236, 303, 233, 119], [325, 79, 88, 263, 369, 357], [84, 153, 130, 191, 168, 186], [271, 34, 18, 36, 347, 318], [143, 307, 160, 179, 352, 49], [301, 143, 116, 223, 259, 170], [172, 228, 111, 183, 369, 197], [302, 53, 346, 29, 289, 152], [276, 309, 227, 303, 376, 1], [366, 13, 359, 63, 45, 298], [307, 370, 375, 143, 175, 140], [307, 187, 130, 136, 373, 372], [89, 113, 119, 216, 317, 311], [111, 160, 291, 126, 56, 377], [6, 182, 255, 284, 209, 66], [233, 65, 124, 53, 339, 220], [99, 337, 87, 33, 63, 153], [40, 90, 334, 383, 388, 130], [283, 128, 144, 397, 41, 125], [217, 31, 295, 164, 185, 135], [116, 91, 267, 331, 86, 126], [357, 176, 310, 219, 159, 358], [318, 399, 188, 155, 349, 242], [197, 271, 377, 236, 343, 375], [192, 51, 100, 229, 73, 363], [395, 388, 139, 97, 158, 238], [131, 376, 340, 392, 245, 296], [125, 86, 100, 120, 18, 89], [264, 17, 198, 352, 309, 68], [302, 148, 31, 1, 157, 241], [141, 223, 363, 106, 318, 228], [154, 165, 142, 321, 238, 232], [260, 270, 243, 268, 280, 124], [38, 271, 288, 135, 309, 42], [303, 119, 356, 307, 193, 349], [307, 107, 139, 201, 182, 32], [89, 361, 122, 266, 151, 366], [199, 97, 169, 217, 6, 36], [7, 342, 156, 267, 19, 311], [345, 38, 155, 107, 102, 141], [197, 356, 54, 78, 129, 241], [104, 72, 335, 22, 358, 280], [279, 247, 289, 274, 52, 316], [56, 196, 285, 258, 391, 65], [229, 236, 344, 184, 394, 21], [370, 254, 168, 118, 112, 36], [118, 270, 262, 328, 184, 40], [91, 93, 1, 387, 133, 379], [352, 11, 227, 286, 371, 220], [80, 224, 317, 331, 244, 311], [390, 359, 394, 242, 283, 32], [119, 217, 360, 293, 129, 122], [357, 283, 372, 335, 312, 230], [288, 238, 87, 264, 267, 385], [385, 204, 256, 351, 32, 375], [31, 264, 145, 120, 203, 249], [103, 58, 60, 150, 292, 69], [49, 152, 80, 392, 320, 298], [290, 6, 303, 168, 201, 378], [279, 354, 387, 121, 146, 153], [264, 136, 223, 300, 283, 29], [310, 264, 93, 292, 86, 280], [375, 86, 80, 161, 100, 253], [298, 189, 192, 233, 143, 333], [130, 106, 39, 298, 378, 373], [34, 90, 393, 371, 168, 60], [394, 195, 134, 49, 88, 132], [204, 275, 308, 364, 293, 29], [82, 150, 300, 79, 342, 43], [72, 221, 389, 8, 32, 324], [37, 381, 35, 252, 250, 315], [188, 126, 361, 219, 395, 250], [119, 234, 169, 353, 151, 66], [57, 135, 152, 164, 49, 231], [161, 234, 55, 84, 42, 10], [245, 233, 380, 363, 114, 81], [251, 214, 79, 341, 160, 201], [352, 124, 165, 274, 61, 307], [3, 1, 135, 349, 250, 55], [380, 10, 299, 5, 192, 71], [77, 66, 282, 62, 121, 103], [287, 156, 323, 144, 268, 195], [125, 263, 187, 36, 193, 357], [352, 316, 337, 323, 44, 283], [109, 58, 5, 23, 100, 95], [205, 119, 101, 369, 155, 322], [391, 128, 248, 115, 359, 382], [112, 390, 315, 55, 301, 19], [188, 276, 222, 256, 277, 226], [187, 40, 84, 212, 275, 221], [41, 155, 228, 343, 229, 232], [138, 150, 303, 308, 307, 202], [358, 100, 88, 281, 355, 24], [167, 68, 141, 379, 38, 241], [123, 294, 252, 17, 147, 227], [44, 393, 112, 270, 38, 155], [133, 249, 383, 106, 4, 399], [234, 339, 142, 335, 97, 67], [135, 283, 21, 288, 382, 237], [388, 386, 394, 267, 3, 227], [258, 238, 164, 146, 166, 178], [265, 327, 295, 233, 162, 134], [271, 115, 371, 261, 322, 151], [319, 337, 48, 277, 390, 290], [232, 19, 141, 55, 383, 287], [111, 56, 262, 159, 293, 148], [326, 191, 83, 253, 228, 133], [293, 173, 350, 316, 223, 67], [316, 123, 137, 115, 13, 242], [95, 189, 192, 223, 26, 287], [187, 320, 48, 223, 206, 149], [165, 162, 260, 257, 389, 346], [22, 187, 373, 1, 268, 204], [335, 72, 135, 277, 201, 250], [159, 391, 393, 270, 155, 174], [260, 96, 182, 27, 186, 154], [211, 109, 25, 48, 156, 349], [174, 217, 66, 314, 85, 210], [114, 45, 143, 183, 7, 372], [280, 278, 61, 320, 257, 377], [213, 142, 118, 134, 60, 323], [69, 156, 244, 334, 363, 1], [354, 346, 135, 171, 164, 389], [134, 260, 315, 128, 198, 207], [84, 190, 18, 372, 373, 110], [386, 340, 273, 367, 135, 204], [331, 102, 361, 273, 21, 165], [219, 177, 253, 163, 320, 152], [345, 259, 23, 49, 317, 258], [224, 268, 382, 248, 214, 303], [53, 112, 391, 155, 199, 25], [34, 336, 142, 211, 48, 352], [16, 55, 6, 352, 368, 218], [396, 231, 283, 174, 312, 69], [199, 20, 54, 65, 154, 244], [326, 123, 387, 305, 90, 245], [197, 194, 156, 318, 338, 85], [166, 240, 320, 26, 318, 385], [251, 136, 214, 3, 45, 368], [386, 257, 101, 34, 144, 19], [150, 260, 85, 211, 124, 128], [189, 86, 261, 373, 92, 387], [287, 362, 148, 205, 58, 367], [2, 210, 311, 359, 253, 334], [234, 201, 62, 57, 385, 360], [263, 101, 388, 167, 210, 187], [19, 217, 168, 163, 212, 52], [306, 312, 223, 193, 364, 246], [151, 207, 393, 281, 330, 345], [243, 117, 12, 366, 77, 99], [167, 264, 304, 197, 250, 301], [177, 117, 62, 189, 381, 331], [325, 308, 22, 270, 67, 213], [277, 246, 271, 17, 302, 141], [160, 273, 66, 252, 148, 260], [159, 135, 133, 370, 59, 242], [378, 201, 246, 242, 360, 158], [219, 116, 30, 44, 59, 194], [241, 97, 3, 330, 243, 194], [224, 82, 7, 309, 162, 25], [251, 108, 172, 151, 96, 114], [6, 391, 345, 205, 124, 328], [221, 240, 294, 305, 23, 280], [361, 209, 164, 152, 1, 252], [89, 360, 57, 318, 306, 342], [347, 30, 134, 154, 321, 313], [14, 73, 356, 284, 241, 168], [96, 384, 211, 24, 313, 132], [288, 93, 54, 111, 240, 229], [18, 141, 307, 64, 286, 23], [83, 211, 157, 288, 209, 24], [350, 53, 263, 87, 277, 345], [199, 61, 365, 66, 194, 161], [87, 351, 195, 188, 399, 173], [380, 88, 131, 397, 386, 48], [79, 266, 379, 396, 325, 169], [314, 84, 27, 312, 281, 82], [49, 106, 189, 279, 30, 235], [204, 18, 354, 279, 9, 263], [115, 130, 225, 92, 331, 22], [57, 379, 362, 68, 222, 154], [146, 248, 230, 115, 187, 243], [318, 388, 184, 311, 13, 116], [257, 149, 89, 297, 352, 93], [110, 181, 247, 47, 39, 352], [351, 312, 177, 204, 353, 69], [271, 43, 245, 133, 230, 212], [51, 266, 277, 32, 196, 284], [91, 388, 115, 15, 104, 57], [324, 172, 122, 31, 155, 242], [98, 250, 271, 76, 361, 135], [282, 10, 29, 47, 113, 75], [20, 159, 50, 257, 209, 132], [4, 397, 264, 144, 396, 301], [27, 384, 208, 183, 74, 170], [203, 342, 96, 76, 193, 47], [359, 287, 111, 151, 268, 292], [388, 87, 175, 146, 350, 171], [246, 377, 278, 352, 142, 150], [99, 15, 273, 270, 330, 150], [189, 164, 302, 351, 240, 49], [398, 154, 146, 24, 184, 46], [258, 387, 78, 41, 92, 245], [90, 395, 61, 134, 367, 25], [128, 369, 376, 264, 157, 58], [116, 211, 111, 378, 216, 80], [256, 373, 228, 193, 254, 399], [17, 58, 395, 311, 138, 53], [114, 169, 15, 119, 278, 281], [374, 273, 245, 26, 360, 136], [382, 274, 223, 200, 356, 190], [187, 333, 213, 64, 5, 143], [135, 301, 64, 344, 377, 217], [93, 351, 175, 219, 239, 286], [212, 107, 283, 98, 307, 37], [342, 244, 391, 388, 301, 163], [265, 188, 43, 79, 305, 111], [79, 196, 312, 42, 22, 299], [391, 167, 20, 382, 218, 99], [8, 298, 147, 329, 158, 195], [235, 369, 187, 135, 253, 265], [365, 311, 243, 361, 298, 35], [170, 277, 332, 48, 178, 359], [39, 309, 252, 342, 57, 291], [102, 110, 280, 56, 3, 284], [56, 390, 196, 243, 23, 12]]\n",
      "[716, 545, 608, 602, 591, 25, 495, 362, 366, 680, 62, 713, 454, 692, 698, 404, 697, 401, 346, 252, 469, 632, 704, 723, 668, 232, 379, 400, 492, 100, 535, 421, 110, 654, 268, 443, 628, 578, 24, 186, 714, 47, 439, 13, 124, 429, 30, 66, 453, 411, 227, 570, 280, 270, 731, 409, 427, 48, 328, 377, 129, 174, 461, 330, 558, 489, 358, 347, 243, 144, 320, 720, 589, 724, 348, 398, 220, 65, 619, 205, 354, 673, 653, 408, 249, 566, 349, 274, 442, 635, 76, 548, 457, 123, 664, 223, 175, 37, 532, 382, 356, 200, 538, 281, 107, 730, 447, 325, 261, 260, 257, 182, 484, 154, 728, 208, 387, 729, 541, 419, 652, 7, 392, 491, 285, 128, 41, 685, 190, 503, 259, 36, 670, 54, 459, 230, 519, 493, 198, 19, 556, 298, 183, 151, 452, 73, 426, 512, 57, 369, 571, 337, 709, 178, 106, 239, 644, 471, 618, 173, 451, 580, 733, 659, 595, 197, 303, 547, 121, 515, 651, 165, 187, 430, 403, 687, 699, 563, 373, 483, 506, 245, 686, 265, 255, 80, 212, 677, 703, 159, 650, 586, 118, 480, 333, 60, 215, 684, 180, 326, 156, 666, 352, 29, 67, 56, 103, 642, 1, 339, 272, 688, 319, 46, 203, 176, 201, 153, 440, 58, 622, 579, 726, 126, 463, 613, 671, 648, 637, 604, 397, 226, 108, 609, 266, 488, 458, 276, 656, 355, 263, 83, 224, 332, 283, 511, 669, 40, 302, 275, 63, 468, 217, 683, 501, 228, 113, 99, 96, 514, 316, 441, 75, 50, 308, 524, 617, 313, 3, 251, 633, 587, 20, 207, 528, 264, 18, 213, 211, 134, 26, 307, 85, 465, 732, 202, 616, 568, 389, 660, 49, 5, 718, 137, 157, 147, 28, 363, 593, 641, 262, 204, 603, 210, 222, 353, 436, 376, 543, 391, 290, 0, 596, 564, 567, 663, 657, 91, 549, 499, 630, 612, 102, 455, 269, 219, 629, 585, 667, 111, 278, 420, 530, 710, 233, 195, 523, 142, 286, 17, 95, 413, 700, 341, 721, 625, 386, 702, 149, 2, 193, 676, 672, 553, 141, 55, 240, 478, 384, 614, 658, 10, 695, 138, 517, 162, 311, 109, 582, 725, 15, 418, 209, 136, 120, 513, 485, 694, 481, 374, 448, 576, 304, 407, 4, 368, 301, 460, 498, 445, 84, 554, 359, 155, 125, 624, 188, 78, 590, 552, 336, 634, 417, 122, 199, 597, 434, 431, 9, 35, 342, 470, 221, 322, 177, 711, 34, 473, 636, 299, 43, 537, 296, 139, 539, 516, 12, 456, 340, 655, 522, 295, 627, 565, 534, 462, 305, 380, 132, 297, 184, 158, 679, 21, 234, 273, 479, 324, 607, 390, 412, 145, 351, 475, 388, 237, 51, 97, 291, 500, 225, 678, 399, 216, 163, 88, 191, 598, 164, 293, 77, 675, 79, 706, 31, 367, 93, 573, 112, 39, 662, 707, 196, 104, 321, 640, 61, 557, 59, 218, 115, 561, 92, 246, 160, 284, 717, 250, 167, 288, 171, 331, 258, 428, 490, 231, 119, 27, 422, 477, 531, 335, 661, 599, 708, 357, 146, 639, 592, 433, 323, 521, 206, 520, 594, 560, 329, 681, 395, 482, 574, 256, 68, 44, 242, 719, 727, 168, 166, 235, 236, 562, 621, 544, 383, 385, 94, 312, 105, 267, 610, 8, 248, 294, 14, 247, 179, 117, 238, 502, 569, 350, 327, 192, 581, 361, 22, 424, 45, 98, 600, 309, 381, 69, 365, 81, 11, 425, 334, 646, 86, 623, 466, 370, 345, 70, 318, 464, 42, 712, 508, 394, 33, 23, 402, 32, 533, 689, 152, 277, 507, 693, 254, 53, 194, 435, 525, 189, 241, 310, 505, 300, 89, 371, 643, 82, 715, 722, 550, 244, 396, 131, 292, 518, 74, 575, 343, 306, 279, 185, 476, 486, 444, 497, 682, 130, 253, 127, 72, 90, 649, 378, 526, 540, 467, 360, 372, 114, 559, 317, 135, 143, 536, 16, 588, 71, 406, 611, 529, 626, 415, 665, 631, 449, 504, 701, 509, 496, 410, 181, 229, 487, 172, 282, 287, 555, 437, 133, 214, 393, 140, 150, 510, 438, 364, 620, 87, 271, 450, 551, 64, 314, 170, 414, 289, 542, 38, 52, 494, 6, 691, 546, 375, 583, 572, 416, 116, 696, 148, 338, 584, 705, 161, 446, 169, 474, 690, 527, 405, 472, 423, 645, 605, 101, 606, 432, 674, 344, 601, 647, 577, 315, 615, 638]\n",
      "[[9, 0, 10, 11, 2, 13], [11, 9, 2, 7, 3, 14], [0, 6, 12, 13, 11, 2], [2, 7, 9, 12, 8, 1], [3, 9, 8, 7, 14, 13], [12, 7, 5, 0, 14, 10], [8, 13, 6, 10, 11, 9], [2, 12, 13, 11, 10, 3], [13, 3, 2, 8, 1, 5], [7, 9, 10, 4, 0, 8], [7, 10, 1, 2, 9, 11], [2, 13, 10, 4, 5, 7], [10, 1, 9, 5, 14, 12], [4, 1, 14, 5, 11, 2], [3, 7, 12, 0, 5, 6], [0, 11, 9, 8, 2, 1], [8, 9, 10, 5, 2, 11], [4, 3, 7, 10, 11, 1], [3, 6, 2, 4, 9, 13], [3, 4, 13, 11, 5, 2], [13, 9, 5, 14, 1, 12], [6, 0, 12, 2, 11, 9], [7, 13, 8, 10, 12, 5], [13, 2, 14, 12, 0, 3], [1, 11, 6, 14, 4, 13], [11, 13, 9, 2, 8, 12], [3, 9, 6, 11, 14, 0], [14, 8, 1, 2, 10, 9], [9, 5, 11, 0, 1, 13], [8, 11, 5, 1, 9, 4], [2, 9, 6, 14, 3, 13], [13, 0, 5, 3, 14, 10], [11, 2, 6, 14, 7, 4], [2, 6, 9, 1, 0, 8], [9, 11, 14, 6, 5, 13], [3, 5, 12, 13, 6, 2], [2, 10, 7, 6, 5, 13], [1, 14, 3, 11, 12, 7], [8, 6, 14, 11, 7, 3], [3, 0, 13, 10, 14, 1], [3, 1, 14, 6, 11, 5], [5, 12, 7, 1, 4, 0], [4, 2, 0, 13, 3, 12], [4, 11, 0, 1, 5, 13], [5, 4, 14, 13, 3, 8], [11, 5, 6, 1, 0, 12], [14, 12, 5, 6, 1, 9], [5, 2, 3, 8, 11, 9], [8, 14, 0, 3, 5, 4], [2, 8, 6, 11, 12, 4], [10, 4, 8, 11, 2, 5], [5, 14, 3, 4, 9, 7], [7, 8, 0, 1, 6, 2], [12, 2, 1, 6, 9, 5], [8, 13, 11, 0, 7, 14], [11, 0, 8, 14, 10, 9], [11, 5, 6, 4, 7, 1], [0, 9, 14, 12, 10, 11], [2, 5, 8, 13, 0, 9], [9, 12, 7, 11, 1, 5], [11, 1, 8, 10, 5, 9], [10, 5, 13, 6, 7, 9], [0, 8, 6, 4, 3, 1], [6, 1, 12, 7, 9, 13], [0, 3, 5, 9, 4, 6], [1, 12, 7, 9, 14, 13], [13, 3, 8, 7, 0, 6], [4, 0, 3, 10, 14, 5], [4, 5, 8, 11, 13, 3], [7, 4, 8, 0, 1, 2], [12, 5, 14, 3, 11, 8], [14, 11, 5, 10, 4, 2], [4, 9, 0, 13, 7, 10], [2, 11, 4, 9, 14, 13], [14, 2, 5, 1, 0, 12], [7, 4, 2, 13, 5, 9], [9, 14, 3, 8, 4, 11], [10, 13, 3, 12, 0, 2], [0, 3, 6, 11, 1, 13], [9, 2, 0, 13, 4, 7], [3, 2, 1, 9, 11, 14], [6, 7, 2, 3, 4, 12], [11, 0, 6, 9, 5, 1], [14, 1, 12, 10, 11, 6], [5, 8, 14, 0, 6, 10], [13, 5, 14, 8, 7, 4], [8, 12, 4, 7, 6, 0], [2, 5, 10, 0, 13, 6], [4, 2, 7, 1, 9, 0], [5, 3, 7, 10, 12, 8], [10, 6, 8, 3, 5, 11], [12, 5, 7, 0, 10, 8], [11, 7, 2, 0, 5, 1], [13, 3, 10, 8, 6, 9], [2, 9, 13, 5, 7, 14], [4, 8, 14, 9, 0, 12], [2, 4, 6, 0, 13, 8], [7, 6, 8, 10, 1, 13], [2, 10, 3, 5, 6, 12], [6, 3, 4, 12, 14, 8], [0, 1, 6, 10, 13, 8], [8, 9, 0, 12, 10, 11], [11, 6, 9, 2, 4, 0], [7, 14, 6, 4, 2, 5], [9, 11, 0, 14, 7, 13], [3, 13, 5, 11, 9, 4], [6, 5, 3, 2, 12, 9], [14, 10, 6, 7, 8, 12], [12, 11, 8, 4, 6, 7], [9, 5, 14, 12, 0, 3], [2, 11, 4, 14, 5, 7], [7, 2, 6, 12, 4, 0], [14, 9, 13, 10, 4, 5], [10, 0, 8, 13, 12, 4], [13, 12, 11, 8, 5, 7], [13, 5, 0, 2, 12, 9], [11, 2, 14, 9, 6, 8], [0, 9, 3, 5, 14, 8], [12, 13, 0, 3, 7, 2], [0, 14, 11, 10, 1, 6], [10, 14, 1, 13, 2, 9], [2, 4, 12, 0, 8, 11], [7, 10, 2, 1, 0, 9], [9, 4, 0, 2, 14, 6], [0, 1, 7, 3, 14, 12], [14, 4, 12, 0, 6, 8], [5, 1, 3, 4, 2, 12], [12, 0, 3, 13, 4, 8], [7, 11, 10, 5, 9, 3], [2, 4, 6, 12, 8, 1], [0, 12, 6, 2, 7, 3], [8, 6, 13, 11, 7, 3], [11, 14, 2, 8, 6, 5], [1, 5, 11, 4, 8, 2], [11, 6, 4, 0, 8, 1], [9, 1, 8, 5, 0, 10], [13, 6, 8, 0, 14, 5], [12, 1, 7, 11, 0, 2], [11, 4, 9, 1, 6, 8], [14, 11, 0, 6, 3, 7], [6, 11, 7, 4, 2, 8], [7, 10, 1, 9, 0, 13], [7, 10, 6, 9, 5, 0], [0, 10, 14, 4, 8, 5], [11, 0, 13, 4, 8, 7], [11, 4, 1, 3, 9, 0], [6, 3, 4, 10, 9, 13], [10, 12, 6, 11, 3, 9], [12, 14, 5, 0, 1, 3], [8, 9, 5, 0, 13, 6], [14, 12, 0, 7, 9, 2], [9, 4, 1, 0, 14, 7], [9, 11, 1, 3, 2, 4], [5, 7, 8, 1, 13, 10], [0, 1, 10, 3, 9, 11], [5, 13, 11, 2, 4, 14], [11, 0, 12, 5, 8, 1], [12, 2, 8, 10, 14, 5], [9, 6, 0, 12, 13, 5], [5, 12, 6, 8, 14, 0], [11, 10, 1, 3, 0, 14], [3, 1, 6, 0, 12, 9], [10, 13, 0, 7, 9, 4], [1, 5, 8, 0, 11, 12], [2, 9, 0, 5, 14, 7], [3, 2, 0, 5, 12, 8], [5, 7, 4, 13, 14, 9], [6, 0, 13, 2, 8, 14], [8, 2, 4, 11, 1, 7], [1, 7, 13, 3, 9, 4], [10, 13, 0, 1, 8, 7], [10, 7, 14, 2, 6, 0], [1, 13, 2, 9, 4, 3], [12, 9, 13, 10, 2, 7], [7, 0, 2, 11, 1, 9], [5, 1, 3, 13, 4, 12], [2, 10, 8, 7, 6, 0], [13, 6, 1, 11, 9, 8], [6, 7, 3, 9, 1, 5], [3, 6, 7, 14, 11, 2], [13, 4, 9, 3, 12, 8], [11, 8, 1, 4, 12, 13], [5, 6, 4, 2, 7, 3], [10, 4, 6, 9, 8, 2], [14, 7, 3, 4, 11, 6], [11, 13, 12, 2, 5, 8], [12, 8, 2, 6, 9, 10], [0, 14, 2, 5, 8, 6], [13, 14, 4, 5, 1, 11], [0, 13, 9, 2, 4, 1], [7, 3, 8, 10, 12, 4], [11, 14, 3, 4, 5, 2], [8, 2, 10, 3, 11, 9], [4, 2, 3, 8, 7, 0], [9, 7, 5, 10, 13, 14], [11, 6, 3, 0, 1, 7], [10, 0, 14, 11, 4, 2], [2, 5, 3, 9, 0, 7], [6, 7, 13, 11, 3, 4], [0, 7, 1, 3, 10, 8], [12, 11, 1, 9, 5, 7], [9, 0, 4, 7, 12, 14], [12, 2, 10, 4, 13, 14], [3, 1, 11, 5, 12, 7], [8, 11, 6, 4, 1, 13], [0, 14, 6, 4, 9, 13], [5, 14, 10, 8, 0, 11], [1, 6, 2, 14, 4, 9], [0, 13, 8, 3, 9, 6], [8, 12, 13, 5, 14, 9], [14, 2, 3, 8, 11, 9], [4, 7, 11, 0, 14, 9], [0, 11, 1, 7, 6, 14], [4, 7, 9, 3, 8, 0], [7, 9, 3, 12, 4, 14], [7, 5, 4, 6, 0, 11], [0, 11, 3, 1, 10, 5], [12, 5, 4, 9, 2, 14], [13, 10, 2, 3, 9, 14], [13, 5, 4, 3, 9, 7], [4, 12, 10, 7, 11, 2], [12, 7, 13, 2, 11, 1], [0, 2, 6, 10, 4, 3], [12, 8, 1, 14, 5, 9], [8, 7, 5, 10, 4, 12], [3, 13, 2, 9, 10, 11], [10, 5, 6, 14, 11, 3], [12, 11, 3, 1, 4, 5], [6, 7, 8, 10, 3, 0], [10, 6, 4, 9, 8, 13], [6, 7, 5, 9, 1, 11], [4, 10, 0, 11, 13, 6], [1, 13, 11, 3, 2, 9], [9, 6, 11, 7, 2, 3], [1, 0, 7, 6, 11, 14], [13, 4, 9, 14, 0, 1], [8, 1, 5, 3, 14, 13], [12, 0, 3, 8, 2, 9], [9, 1, 14, 8, 6, 3], [5, 13, 10, 11, 3, 7], [2, 7, 3, 13, 4, 10], [8, 0, 2, 3, 10, 9], [10, 6, 1, 4, 2, 8], [13, 0, 12, 3, 1, 14], [9, 10, 12, 4, 7, 3], [12, 13, 8, 7, 9, 11], [10, 6, 8, 4, 3, 7], [2, 14, 4, 8, 13, 12], [4, 3, 12, 13, 0, 10], [13, 7, 1, 11, 8, 3], [5, 7, 13, 11, 0, 1], [14, 12, 2, 7, 10, 8], [8, 6, 3, 9, 11, 10], [8, 3, 2, 1, 7, 0], [12, 0, 9, 4, 3, 6], [8, 10, 6, 3, 14, 7], [5, 8, 1, 7, 10, 12], [12, 13, 2, 9, 3, 1], [2, 8, 5, 12, 11, 6], [5, 9, 2, 6, 7, 10], [11, 12, 9, 1, 4, 8], [12, 0, 6, 1, 7, 8], [6, 7, 4, 5, 12, 13], [1, 2, 12, 5, 14, 7], [9, 8, 13, 1, 7, 4], [13, 9, 11, 6, 0, 8], [4, 3, 12, 0, 1, 7], [9, 5, 7, 3, 14, 12], [8, 12, 11, 10, 0, 5], [5, 1, 9, 6, 3, 13], [1, 12, 7, 8, 11, 6], [6, 12, 7, 0, 14, 10], [6, 0, 5, 4, 13, 9], [12, 7, 1, 11, 14, 6], [6, 5, 8, 1, 11, 7], [3, 7, 1, 9, 14, 2], [6, 13, 0, 9, 10, 4], [3, 2, 13, 1, 10, 9], [12, 7, 0, 2, 8, 14], [9, 4, 2, 8, 14, 5], [8, 9, 3, 12, 13, 0], [1, 10, 5, 4, 9, 6], [11, 4, 14, 9, 0, 3], [12, 6, 9, 13, 5, 10], [5, 11, 14, 4, 13, 2], [8, 1, 13, 5, 2, 4], [14, 6, 1, 0, 8, 7], [12, 0, 4, 9, 8, 14], [0, 4, 14, 3, 5, 10], [7, 8, 1, 14, 3, 6], [4, 1, 13, 5, 3, 11], [14, 13, 3, 0, 4, 6], [10, 11, 3, 5, 12, 1], [11, 6, 3, 7, 13, 9], [5, 10, 6, 9, 3, 11], [10, 3, 2, 7, 8, 5], [1, 7, 5, 9, 11, 13], [11, 8, 14, 6, 1, 10], [7, 10, 0, 11, 3, 8], [1, 11, 2, 12, 0, 10], [1, 9, 12, 6, 2, 0], [4, 12, 0, 7, 13, 9], [11, 3, 2, 9, 0, 1], [14, 2, 0, 1, 11, 10], [12, 8, 10, 3, 5, 9], [0, 5, 2, 13, 9, 3], [8, 5, 2, 11, 6, 4], [12, 8, 2, 9, 5, 1], [13, 8, 6, 2, 14, 9], [0, 13, 5, 12, 7, 14], [6, 13, 0, 14, 11, 4], [9, 1, 6, 8, 12, 10], [14, 4, 9, 10, 6, 11], [13, 10, 1, 2, 3, 14], [8, 13, 7, 9, 5, 12], [3, 8, 14, 9, 4, 1], [7, 4, 11, 14, 10, 12], [13, 14, 0, 4, 10, 12], [14, 2, 10, 3, 6, 12], [6, 9, 5, 11, 0, 13], [3, 7, 6, 2, 11, 4], [5, 6, 11, 1, 8, 4], [2, 1, 0, 4, 3, 13], [6, 13, 4, 1, 10, 11], [12, 0, 8, 4, 9, 5], [12, 10, 1, 8, 14, 7], [12, 9, 1, 10, 6, 11], [11, 12, 3, 9, 7, 1], [8, 9, 7, 4, 11, 6], [0, 2, 8, 9, 7, 3], [14, 2, 9, 11, 5, 8], [8, 0, 6, 3, 5, 2], [9, 8, 12, 6, 13, 3], [0, 3, 9, 6, 10, 7], [2, 3, 1, 0, 13, 9], [12, 10, 4, 3, 11, 9], [3, 10, 7, 8, 4, 1], [1, 6, 11, 9, 4, 3], [6, 7, 4, 12, 1, 5], [2, 12, 14, 4, 3, 5], [4, 0, 7, 9, 8, 13], [5, 4, 12, 10, 3, 13], [4, 9, 7, 14, 1, 11], [7, 3, 5, 8, 13, 12], [1, 3, 7, 5, 9, 14], [4, 9, 8, 5, 2, 6], [0, 14, 11, 3, 4, 9], [11, 13, 5, 8, 1, 9], [10, 9, 7, 11, 13, 0], [9, 6, 12, 2, 13, 5], [0, 6, 2, 13, 14, 4], [2, 12, 9, 13, 6, 7], [6, 12, 2, 14, 13, 1], [3, 4, 0, 10, 1, 8], [11, 9, 3, 1, 12, 7], [10, 9, 14, 5, 8, 3], [1, 5, 6, 13, 2, 11], [1, 9, 4, 6, 5, 14], [3, 2, 6, 0, 9, 12], [14, 3, 11, 13, 8, 5], [3, 8, 11, 13, 4, 5], [10, 0, 14, 5, 1, 3], [9, 3, 13, 5, 6, 12], [14, 12, 6, 2, 5, 11], [7, 5, 14, 1, 9, 13], [10, 8, 1, 2, 14, 13], [1, 9, 10, 3, 0, 4], [11, 6, 13, 1, 5, 2], [3, 12, 13, 11, 0, 7], [3, 2, 9, 7, 1, 14], [8, 3, 1, 2, 0, 7], [8, 3, 6, 9, 1, 4], [5, 12, 4, 6, 0, 8], [2, 4, 7, 9, 8, 3], [10, 4, 7, 12, 8, 13], [2, 3, 10, 4, 8, 5], [6, 1, 12, 13, 8, 14], [2, 4, 6, 3, 11, 1], [12, 2, 9, 7, 11, 13], [11, 0, 8, 3, 1, 9], [2, 9, 4, 7, 3, 13], [8, 10, 14, 9, 11, 6], [4, 0, 1, 9, 13, 8], [0, 12, 9, 4, 10, 14], [12, 3, 8, 5, 11, 10], [1, 9, 3, 11, 4, 13], [13, 0, 12, 9, 3, 7], [12, 13, 8, 9, 2, 1], [14, 9, 5, 2, 1, 3], [3, 14, 5, 12, 11, 4], [2, 11, 10, 1, 4, 14], [13, 8, 9, 10, 5, 2], [6, 8, 4, 3, 11, 12], [14, 1, 3, 8, 10, 12], [7, 14, 12, 3, 10, 2], [4, 11, 10, 3, 1, 14], [3, 13, 10, 5, 9, 14], [14, 4, 9, 3, 8, 7], [0, 13, 1, 11, 8, 6], [8, 10, 3, 6, 12, 11], [10, 9, 3, 11, 7, 2], [5, 6, 9, 14, 10, 4], [9, 2, 4, 7, 3, 8], [7, 3, 9, 1, 8, 0], [0, 6, 14, 13, 10, 5], [9, 5, 14, 8, 6, 7], [5, 2, 7, 14, 4, 6], [1, 2, 5, 7, 0, 14], [6, 7, 1, 13, 5, 11], [3, 14, 0, 6, 9, 12], [4, 6, 11, 3, 8, 13], [4, 8, 11, 3, 1, 9], [11, 2, 10, 0, 7, 5], [3, 13, 1, 8, 7, 4], [13, 10, 3, 14, 1, 5], [3, 11, 10, 5, 14, 9], [0, 3, 14, 13, 10, 8], [1, 7, 11, 0, 12, 5], [6, 1, 11, 7, 14, 4], [13, 11, 5, 6, 9, 12], [12, 5, 14, 9, 11, 2], [1, 13, 8, 0, 10, 12], [13, 6, 14, 4, 10, 8], [5, 0, 14, 13, 10, 6], [12, 8, 5, 2, 9, 10], [7, 0, 8, 13, 9, 5], [7, 3, 14, 12, 1, 10], [13, 1, 9, 4, 2, 6], [1, 6, 14, 13, 2, 3], [6, 13, 3, 7, 2, 5], [13, 3, 10, 14, 0, 7], [11, 13, 1, 3, 10, 6], [13, 9, 5, 10, 8, 0], [13, 2, 11, 14, 1, 9], [12, 11, 3, 1, 2, 8], [5, 11, 2, 10, 3, 12], [10, 12, 3, 5, 8, 1], [11, 4, 1, 9, 5, 7], [2, 14, 13, 6, 5, 3], [1, 13, 8, 11, 7, 0], [7, 2, 6, 9, 13, 0], [3, 6, 12, 11, 10, 5], [5, 9, 10, 7, 6, 3], [2, 8, 1, 4, 14, 6], [8, 3, 11, 4, 10, 14], [2, 11, 7, 13, 4, 12], [6, 11, 8, 12, 3, 1], [7, 1, 0, 11, 8, 9], [8, 2, 3, 11, 12, 0], [6, 3, 7, 13, 5, 12], [10, 14, 9, 0, 3, 5], [14, 6, 3, 8, 11, 9], [6, 2, 8, 12, 9, 3], [11, 13, 12, 6, 3, 9], [10, 13, 2, 0, 12, 5], [5, 9, 3, 4, 10, 2], [3, 14, 9, 7, 4, 11], [4, 5, 6, 11, 8, 7], [1, 4, 0, 8, 11, 7], [9, 8, 0, 4, 1, 10], [7, 14, 9, 0, 6, 2], [3, 0, 6, 2, 10, 5], [13, 14, 9, 7, 3, 2], [7, 5, 12, 0, 10, 1], [2, 7, 14, 4, 10, 6], [11, 3, 9, 4, 6, 1], [10, 9, 8, 1, 4, 0], [8, 14, 9, 2, 1, 13], [13, 12, 2, 8, 4, 3], [6, 4, 13, 11, 12, 9], [7, 8, 1, 11, 2, 10], [12, 3, 4, 0, 8, 11], [5, 6, 7, 12, 13, 9], [3, 4, 5, 8, 13, 7], [4, 13, 8, 9, 3, 5], [2, 6, 10, 4, 9, 7], [14, 11, 5, 13, 12, 10], [12, 1, 10, 6, 5, 8], [11, 10, 3, 8, 1, 4], [8, 0, 4, 1, 13, 6], [12, 6, 2, 1, 4, 11], [10, 7, 1, 8, 14, 5], [1, 10, 9, 14, 3, 5], [4, 2, 12, 7, 13, 8], [4, 11, 2, 3, 12, 14], [13, 3, 9, 4, 2, 8], [12, 0, 13, 8, 9, 6], [4, 5, 13, 6, 0, 10], [7, 2, 11, 4, 6, 3], [4, 2, 8, 0, 5, 1], [12, 3, 13, 11, 10, 0], [13, 5, 9, 6, 3, 4], [5, 9, 4, 8, 10, 1], [14, 11, 9, 8, 6, 12], [9, 1, 6, 11, 4, 0], [5, 13, 4, 12, 8, 11], [8, 14, 10, 1, 7, 0], [7, 4, 5, 2, 6, 1], [1, 2, 5, 8, 13, 12], [5, 4, 11, 6, 8, 13], [9, 4, 13, 0, 10, 14], [14, 11, 7, 0, 13, 2], [13, 14, 8, 12, 0, 1], [6, 7, 4, 5, 12, 11], [3, 14, 1, 6, 13, 5], [13, 4, 3, 1, 11, 6], [6, 9, 11, 2, 13, 10], [8, 4, 0, 13, 11, 12], [1, 10, 5, 14, 11, 12], [14, 9, 11, 2, 6, 13], [14, 9, 3, 10, 8, 0], [9, 5, 11, 8, 3, 14], [9, 6, 2, 3, 5, 11], [6, 10, 5, 1, 11, 4], [12, 2, 1, 0, 8, 13], [10, 5, 8, 12, 4, 7], [2, 14, 11, 1, 8, 9], [2, 12, 7, 1, 14, 3], [10, 13, 6, 0, 8, 5], [11, 12, 13, 9, 14, 1], [14, 2, 9, 7, 1, 11], [5, 8, 13, 12, 6, 3], [0, 10, 4, 5, 1, 13], [2, 1, 8, 10, 0, 9], [14, 10, 13, 12, 8, 9], [11, 14, 13, 10, 8, 5], [6, 4, 0, 10, 9, 14], [1, 2, 12, 10, 7, 9], [6, 3, 0, 7, 5, 13], [4, 3, 14, 7, 11, 1], [0, 14, 2, 9, 1, 12], [14, 13, 0, 2, 3, 8], [9, 1, 12, 14, 5, 7], [4, 8, 9, 7, 0, 2], [14, 6, 2, 3, 11, 9], [4, 0, 10, 2, 3, 7], [6, 8, 2, 12, 1, 11], [8, 12, 7, 9, 6, 14], [0, 3, 7, 14, 5, 6], [8, 12, 2, 4, 10, 0], [7, 0, 3, 10, 12, 11], [9, 3, 7, 2, 0, 11], [12, 3, 8, 11, 1, 7], [6, 7, 10, 5, 0, 3], [11, 4, 0, 2, 5, 3], [0, 10, 14, 4, 7, 3], [7, 10, 0, 5, 8, 2], [13, 0, 6, 2, 5, 8], [0, 14, 11, 10, 1, 4], [7, 11, 8, 1, 0, 2], [12, 13, 8, 4, 2, 1], [4, 1, 14, 9, 6, 3], [6, 9, 13, 7, 8, 3], [5, 8, 13, 10, 12, 14], [2, 0, 12, 8, 14, 11], [11, 5, 13, 10, 9, 6], [13, 0, 8, 12, 4, 7], [12, 13, 10, 4, 7, 2], [10, 13, 11, 4, 2, 7], [2, 9, 13, 7, 1, 8], [1, 8, 3, 7, 5, 11], [1, 13, 12, 0, 14, 9], [9, 0, 7, 2, 5, 14], [6, 2, 0, 11, 4, 1], [7, 0, 6, 13, 10, 9], [7, 9, 13, 10, 1, 12], [13, 0, 5, 8, 4, 1], [14, 5, 3, 8, 9, 2], [2, 9, 7, 14, 12, 6], [7, 11, 3, 13, 1, 2], [9, 8, 12, 2, 14, 6], [11, 6, 1, 12, 13, 14], [8, 14, 5, 6, 12, 9], [5, 14, 4, 9, 7, 13], [2, 6, 11, 9, 7, 10], [11, 1, 7, 0, 13, 5], [9, 7, 14, 3, 6, 2], [2, 4, 6, 10, 0, 13], [3, 5, 4, 0, 9, 10], [12, 3, 0, 6, 4, 5], [6, 1, 4, 8, 9, 11], [12, 8, 2, 11, 13, 0], [4, 12, 7, 14, 0, 10], [8, 6, 2, 12, 11, 4], [2, 4, 6, 1, 13, 14], [0, 1, 14, 6, 4, 5], [8, 9, 6, 2, 12, 10], [9, 4, 6, 1, 8, 12], [12, 9, 5, 10, 13, 3], [1, 2, 6, 14, 8, 5], [0, 1, 5, 2, 3, 4], [14, 4, 2, 0, 7, 9], [13, 3, 1, 2, 9, 0], [5, 14, 1, 12, 6, 3], [11, 14, 6, 4, 10, 3], [2, 13, 5, 12, 9, 6], [9, 6, 7, 2, 4, 12], [14, 8, 12, 1, 2, 9], [7, 1, 9, 8, 0, 13], [0, 4, 11, 3, 12, 7], [2, 3, 1, 7, 0, 12], [14, 7, 3, 10, 11, 0], [7, 6, 10, 14, 3, 1], [13, 6, 14, 1, 3, 10], [4, 0, 2, 3, 13, 11], [10, 0, 8, 13, 3, 6], [9, 6, 5, 0, 8, 7], [6, 8, 2, 7, 12, 13], [5, 10, 6, 7, 12, 13], [5, 13, 8, 1, 9, 10], [0, 4, 11, 7, 14, 1], [0, 12, 2, 4, 7, 9], [9, 3, 14, 6, 11, 7], [6, 14, 7, 12, 1, 2], [13, 3, 2, 4, 14, 6], [14, 4, 9, 11, 2, 12], [6, 3, 5, 11, 2, 9], [9, 0, 4, 1, 13, 3], [2, 13, 7, 8, 1, 0], [8, 2, 7, 5, 11, 10], [6, 13, 11, 10, 9, 0], [13, 7, 5, 9, 4, 0], [8, 11, 13, 5, 4, 3], [9, 7, 14, 3, 1, 10], [7, 3, 6, 8, 0, 1], [10, 11, 12, 13, 0, 6], [6, 0, 5, 4, 13, 3], [7, 4, 9, 14, 3, 2], [8, 14, 6, 9, 3, 7], [8, 0, 10, 2, 4, 9], [6, 3, 12, 4, 0, 11], [13, 9, 5, 7, 2, 11], [9, 12, 1, 4, 6, 7], [5, 10, 3, 6, 12, 0], [11, 1, 12, 13, 2, 6], [2, 14, 12, 10, 1, 13], [4, 11, 2, 12, 5, 6], [1, 6, 14, 10, 13, 0], [7, 13, 2, 4, 12, 5], [8, 1, 10, 2, 7, 3], [7, 1, 2, 12, 5, 11], [8, 11, 3, 4, 6, 1], [4, 11, 7, 3, 13, 6], [4, 5, 13, 6, 14, 12], [0, 2, 11, 10, 8, 7], [3, 5, 6, 8, 4, 12], [9, 10, 12, 3, 13, 4], [14, 10, 3, 7, 6, 9], [3, 14, 6, 1, 13, 5], [10, 13, 9, 6, 14, 2], [12, 5, 3, 6, 14, 1], [9, 5, 7, 0, 11, 13], [10, 6, 14, 0, 11, 3], [13, 14, 2, 10, 6, 4], [10, 8, 14, 9, 12, 2], [12, 8, 10, 4, 14, 11], [12, 10, 2, 3, 4, 9], [12, 11, 7, 13, 6, 4], [4, 3, 10, 8, 9, 1], [6, 0, 10, 3, 9, 1], [12, 14, 2, 6, 1, 11], [12, 8, 7, 9, 10, 3], [11, 5, 2, 14, 6, 10], [5, 3, 12, 6, 0, 10], [9, 14, 11, 5, 1, 2], [9, 6, 5, 8, 10, 4], [11, 2, 10, 12, 13, 7], [7, 9, 4, 1, 13, 12], [1, 4, 14, 2, 9, 7], [13, 14, 2, 0, 9, 6], [2, 10, 13, 6, 8, 9], [1, 14, 5, 2, 10, 11], [0, 12, 14, 8, 9, 1], [9, 12, 10, 5, 2, 11], [10, 3, 1, 6, 8, 12], [11, 4, 5, 9, 2, 6], [1, 12, 3, 8, 7, 9], [12, 10, 13, 3, 14, 4], [8, 9, 6, 7, 1, 0], [13, 5, 3, 9, 10, 7], [10, 7, 3, 5, 9, 0], [13, 0, 5, 8, 4, 11], [1, 12, 5, 11, 9, 4], [13, 4, 11, 5, 2, 10], [11, 0, 5, 2, 10, 1], [3, 1, 4, 14, 10, 12], [11, 7, 8, 5, 9, 12], [5, 2, 8, 6, 7, 12], [7, 8, 2, 12, 14, 10], [8, 4, 12, 5, 1, 9], [12, 2, 4, 5, 6, 13], [11, 9, 5, 0, 12, 1], [11, 1, 8, 10, 4, 12], [4, 0, 11, 10, 1, 2], [2, 14, 1, 8, 5, 10], [13, 6, 10, 4, 11, 2], [11, 14, 1, 0, 6, 4], [3, 4, 13, 14, 7, 5], [7, 8, 9, 13, 12, 5], [14, 5, 7, 2, 6, 8], [12, 8, 14, 10, 2, 5], [0, 8, 10, 11, 9, 2], [3, 13, 2, 9, 12, 8], [5, 0, 12, 2, 10, 3], [1, 2, 5, 7, 3, 11], [1, 5, 11, 7, 8, 10], [7, 11, 13, 6, 8, 0], [14, 6, 9, 4, 10, 8], [5, 11, 14, 4, 13, 10], [14, 13, 5, 7, 9, 3], [4, 0, 11, 14, 13, 10], [5, 4, 7, 10, 3, 13], [11, 3, 1, 4, 13, 10], [10, 4, 14, 6, 11, 7], [13, 12, 14, 5, 7, 11], [12, 1, 6, 0, 5, 9], [5, 10, 1, 8, 2, 13], [3, 11, 9, 13, 10, 4], [1, 3, 11, 13, 12, 9], [3, 13, 12, 4, 10, 2], [5, 8, 9, 14, 6, 4], [1, 0, 12, 13, 5, 9], [1, 12, 5, 2, 6, 13], [5, 10, 6, 14, 4, 7], [3, 12, 2, 1, 11, 9], [4, 12, 14, 1, 10, 7], [5, 0, 3, 8, 10, 2], [7, 5, 13, 14, 2, 9], [1, 12, 6, 0, 13, 10], [3, 2, 4, 12, 14, 13], [9, 12, 10, 1, 13, 6], [14, 13, 3, 2, 8, 10], [5, 1, 0, 9, 4, 3], [14, 4, 7, 8, 3, 11]]\n"
     ]
    }
   ],
   "source": [
    "# data_dir=\"../../datasets/amazon_google_exp/\"\n",
    "# data_name=\"amazon_google_exp\"\n",
    "# src_orig_file=f\"{data_dir}/source.csv\"\n",
    "# tgt_orig_file=f\"{data_dir}/target.csv\"\n",
    "# golden_mappings=f\"{data_dir}/groundtruth.csv\"\n",
    "\n",
    "\n",
    "data_dir = \"../../datasets/gdc_all/\"\n",
    "def parse_data_dir_gdc(data_dir):\n",
    "    \n",
    "    data_name = data_dir\n",
    "    src_orig_file = pd.read_csv(os.path.join(data_dir, \"source.csv\"))\n",
    "    tgt_orig_file = pd.read_csv(\"../../resource/gdc_table.csv\")\n",
    "    # gt = pd.read_csv(\"../../datasets/dou/groundtruth.csv\")\n",
    "    golden_mappings=os.path.join(data_dir, \"groundtruth.csv\")\n",
    "    return data_name, src_orig_file, tgt_orig_file, golden_mappings\n",
    "\n",
    "data_name, src_orig_file, tgt_orig_file, golden_mappings = parse_data_dir_gdc(data_dir)\n",
    "\n",
    "isresmat = TrainApp(\n",
    "    n_trn_cols=200,\n",
    "    batch_size=1,\n",
    "    frag_height=6,\n",
    "    frag_width=12,\n",
    "    learning_rate=3e-5,\n",
    "    col_name_prob=0,\n",
    "    store_matches=0,\n",
    "    comment=\"inst_001\",\n",
    "    dataset_name=data_name,\n",
    "    orig_file_src=src_orig_file,\n",
    "    orig_file_tgt=tgt_orig_file,\n",
    "    orig_file_golden_matches=golden_mappings,\n",
    "    process_mode=1,\n",
    "    n_val_cols=1,\n",
    "    model_checkpoint=\"checkpoint/checkpoint.pt\",\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4f348e8c-918a-45bd-931c-c8e3e53366d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 6116/6117 [29:31<00:00,  3.68it/s]IOPub data rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n",
      "2024-10-11 02:30:10 CRITICAL top_ground_truth_matches----------->[frozenset({('target', 'ensat_pathologic_stage'), ('source', 'tumor_Stage-Pathological')}), frozenset({('source', 'Gender'), ('target', 'relationship_gender')}), frozenset({('target', 'relationship_gender'), ('source', 'gender')}), frozenset({('target', 'ann_arbor_pathologic_stage'), ('source', 'tumor_Stage-Pathological')}), frozenset({('target', 'methylation_beta_values'), ('source', 'NMF.Cluster.Membership.Score')}), frozenset({('target', 'ensat_pathologic_stage'), ('source', 'tumor_stage')}), frozenset({('target', 'ann_arbor_pathologic_stage'), ('source', 'tumor_stage')}), frozenset({('target', 'annotated_somatic_mutations'), ('source', 'CDKN2A.pathway.alteration')}), frozenset({('source', 'excluded_in_downstream_analysis'), ('target', 'awg_review')}), frozenset({('target', 'ensat_pathologic_stage'), ('source', 'Tumor_Stage_Pathological')}), frozenset({('target', 'ann_arbor_clinical_stage'), ('source', 'tumor_stage')}), frozenset({('target', 'somatic_mutation_indexes'), ('source', 'CDKN2A.pathway.alteration')}), frozenset({('target', 'ann_arbor_clinical_stage'), ('source', 'tumor_Stage-Pathological')}), frozenset({('source', 'excluded_in_downstream_analysis'), ('target', 'selected_normal_low_pass_wgs')}), frozenset({('source', 'Gender'), ('target', 'gender')}), frozenset({('target', 'weight'), ('source', 'weight')}), frozenset({('target', 'ann_arbor_pathologic_stage'), ('source', 'Tumor_Stage_Pathological')}), frozenset({('target', 'ann_arbor_clinical_stage'), ('source', 'Tumor_Stage_Pathological')}), frozenset({('target', 'ovarian_surface_involvement'), ('source', 'perineural_invasion')}), frozenset({('target', 'uicc_clinical_m'), ('source', 'FIGO_stage')}), frozenset({('target', 'cog_renal_stage'), ('source', 'tumor_stage')}), frozenset({('target', 'gender'), ('source', 'gender')}), frozenset({('target', 'masked_somatic_mutations'), ('source', 'CDKN2A.pathway.alteration')}), frozenset({('source', 'BMI'), ('target', 'bmi')}), frozenset({('source', 'tumor_laterality'), ('target', 'undescended_testis_corrected_laterality')}), frozenset({('target', 'cog_renal_stage'), ('source', 'tumor_Stage-Pathological')}), frozenset({('source', 'tumor_laterality'), ('target', 'undescended_testis_history_laterality')}), frozenset({('source', 'tumor_focality'), ('target', 'tumor_focality')}), frozenset({('target', 'relationship_gender'), ('source', 'sex')}), frozenset({('target', 'ajcc_pathologic_m'), ('source', 'FIGO_stage')}), frozenset({('target', 'cog_renal_stage'), ('source', 'Tumor_Stage_Pathological')}), frozenset({('target', 'ajcc_clinical_m'), ('source', 'FIGO_stage')}), frozenset({('target', 'submitted_unaligned_reads_files'), ('source', 'ESTIMATEScore')}), frozenset({('target', 'cog_liver_stage'), ('source', 'tumor_Stage-Pathological')}), frozenset({('source', 'tumor_laterality'), ('target', 'biospecimen_laterality')}), frozenset({('target', 'cog_liver_stage'), ('source', 'tumor_stage')}), frozenset({('target', 'uicc_pathologic_m'), ('source', 'FIGO_stage')}), frozenset({('source', 'excluded_in_downstream_analysis'), ('target', 'no_matched_normal_wgs')}), frozenset({('target', 'gender'), ('source', 'sex')}), frozenset({('target', 'cog_liver_stage'), ('source', 'Tumor_Stage_Pathological')}), frozenset({('source', 'StromalScore'), ('target', 'submitted_unaligned_reads_files')}), frozenset({('source', 'excluded_in_downstream_analysis'), ('target', 'age_is_obfuscated')}), frozenset({('source', 'Tumor_Focality'), ('target', 'tumor_focality')}), frozenset({('source', 'excluded_in_downstream_analysis'), ('target', 'extrascleral_extension')}), frozenset({('target', 'tumor_grade'), ('source', 'Grade')}), frozenset({('source', 'ERBB2.Proteogenomic.Status'), ('target', 'peritoneal_washing_results')})]\n",
      "100%|| 6117/6117 [29:34<00:00,  3.45it/s]\n"
     ]
    }
   ],
   "source": [
    "isresmat.do_trn(0)\n",
    "# isresmat.eval_to_match()\n",
    "# isresmat.get_matches(top_k=2)\n",
    "# isresmat.update_curr_match_status()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cd24b2a3-5fac-468e-8f0c-36c785c51cf3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import torch\n",
    "\n",
    "torch.save(isresmat.model.state_dict(), \"checkpoint/checkpoint.pt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c5b0a83a-7c33-4def-9c0e-8ee3fbd3d42a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BertFoMatching(\n",
       "  (bert): BertModel(\n",
       "    (embeddings): BertEmbeddings(\n",
       "      (word_embeddings): Embedding(30522, 768, padding_idx=0)\n",
       "      (position_embeddings): Embedding(512, 768)\n",
       "      (token_type_embeddings): Embedding(2, 768)\n",
       "      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "      (dropout): Dropout(p=0.1, inplace=False)\n",
       "    )\n",
       "    (encoder): BertEncoder(\n",
       "      (layer): ModuleList(\n",
       "        (0-11): 12 x BertLayer(\n",
       "          (attention): BertAttention(\n",
       "            (self): BertSdpaSelfAttention(\n",
       "              (query): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (key): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (value): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "            (output): BertSelfOutput(\n",
       "              (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "          )\n",
       "          (intermediate): BertIntermediate(\n",
       "            (dense): Linear(in_features=768, out_features=3072, bias=True)\n",
       "            (intermediate_act_fn): GELUActivation()\n",
       "          )\n",
       "          (output): BertOutput(\n",
       "            (dense): Linear(in_features=3072, out_features=768, bias=True)\n",
       "            (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "            (dropout): Dropout(p=0.1, inplace=False)\n",
       "          )\n",
       "        )\n",
       "      )\n",
       "    )\n",
       "    (pooler): BertPooler(\n",
       "      (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "      (activation): Tanh()\n",
       "    )\n",
       "  )\n",
       "  (dropout): Dropout(p=0.1, inplace=False)\n",
       "  (projector): Sequential(\n",
       "    (0): Linear(in_features=768, out_features=512, bias=True)\n",
       "    (1): ReLU()\n",
       "    (2): Linear(in_features=512, out_features=512, bias=True)\n",
       "  )\n",
       "  (src_agents_layer): AgentsLayer()\n",
       "  (tgt_agents_layer): AgentsLayer()\n",
       "  (CE_loss_func): CrossEntropyLoss()\n",
       "  (KL_loss_func): KLDivLoss()\n",
       ")"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "isresmat.model.load_state_dict(torch.load(\"checkpoint/checkpoint.pt\", weights_only=True))\n",
    "isresmat.model.eval()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ddb655f0-e7dd-48cb-a776-d5185b0d7b58",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-10-11 04:48:52 INFO begin the eval-to-match process\n",
      "100%|| 12/12 [00:08<00:00,  1.48it/s]\n",
      "100%|| 31/31 [00:04<00:00,  7.01it/s]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{(('source', 'HPV_inference'),\n",
       "  ('target', 'barretts_esophagus_goblet_cells_present')): 1.0000001192092896,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'clinical_trial_indicator')): 1.0000001192092896,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'sarcomatoid_present')): 1.0000001192092896,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'relative_deceased')): 0.9999992251396179,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'no_matched_normal_wgs')): 0.9999856948852539,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'age_is_obfuscated')): 0.9999833106994629,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'request_submission')): 0.9999833106994629,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'selected_normal_low_pass_wgs')): 0.9999786615371704,\n",
       " (('source', 'tumor_stage'),\n",
       "  ('target', 'ensat_pathologic_stage')): 0.9999632239341736,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'diagnosis_is_primary_disease')): 0.9999560117721558,\n",
       " (('source', 'tumor_Stage-Pathological'),\n",
       "  ('target', 'ensat_pathologic_stage')): 0.9999451637268066,\n",
       " (('source', 'tumor_Stage-Pathological'),\n",
       "  ('target', 'ann_arbor_pathologic_stage')): 0.999893307685852,\n",
       " (('source', 'tumor_Stage-Pathological'),\n",
       "  ('target', 'ann_arbor_clinical_stage')): 0.9998637437820435,\n",
       " (('source', 'tumor_stage'),\n",
       "  ('target', 'ann_arbor_pathologic_stage')): 0.9998149871826172,\n",
       " (('source', 'tumor_stage'),\n",
       "  ('target', 'ann_arbor_clinical_stage')): 0.9997552633285522,\n",
       " (('source', 'tumor_laterality'),\n",
       "  ('target', 'undescended_testis_corrected_laterality')): 0.9997355937957764,\n",
       " (('source', 'Gender'), ('target', 'relationship_gender')): 0.9996416568756104,\n",
       " (('source', 'sex'), ('target', 'relationship_gender')): 0.9996389150619507,\n",
       " (('source', 'Tumor_Stage_Pathological'),\n",
       "  ('target', 'ann_arbor_pathologic_stage')): 0.9996256828308105,\n",
       " (('source', 'gender'), ('target', 'relationship_gender')): 0.9996203184127808,\n",
       " (('source', 'Tumor_Stage_Pathological'),\n",
       "  ('target', 'ann_arbor_clinical_stage')): 0.9995628595352173,\n",
       " (('source', 'tumor_focality'),\n",
       "  ('target', 'tumor_focality')): 0.999444842338562,\n",
       " (('source', 'Tumor_Stage_Pathological'),\n",
       "  ('target', 'ensat_pathologic_stage')): 0.999334454536438,\n",
       " (('source', 'gender'), ('target', 'gender')): 0.9991286993026733,\n",
       " (('source', 'Tumor_Focality'),\n",
       "  ('target', 'tumor_focality')): 0.9991159439086914,\n",
       " (('source', 'Gender'), ('target', 'gender')): 0.9990547895431519,\n",
       " (('source', 'sex'), ('target', 'gender')): 0.9989615678787231,\n",
       " (('source', 'perineural_invasion'),\n",
       "  ('target', 'satellite_nodule_present')): 0.9980735778808594,\n",
       " (('source', 'CDKN2A.pathway.alteration'),\n",
       "  ('target', 'annotated_somatic_mutations')): 0.9972738027572632,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'relative_deceased')): 0.9971959590911865,\n",
       " (('source', 'Tumor_Stage_Pathological'),\n",
       "  ('target', 'cog_renal_stage')): 0.9971829652786255,\n",
       " (('source', 'lymph_vascular_invasion'),\n",
       "  ('target', 'ovarian_surface_involvement')): 0.9971659183502197,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'relative_deceased')): 0.9970887899398804,\n",
       " (('source', 'Grade'), ('target', 'who_nte_grade')): 0.9970269203186035,\n",
       " (('source', 'Tumor_Stage_Pathological'),\n",
       "  ('target', 'cog_liver_stage')): 0.9968781471252441,\n",
       " (('source', 'tumor_laterality'),\n",
       "  ('target', 'undescended_testis_history_laterality')): 0.9968085885047913,\n",
       " (('source', 'lymph_vascular_invasion'),\n",
       "  ('target', 'satellite_nodule_present')): 0.9967535734176636,\n",
       " (('source', 'tumor_Stage-Pathological'),\n",
       "  ('target', 'cog_renal_stage')): 0.9967226982116699,\n",
       " (('source', 'tumor_stage'),\n",
       "  ('target', 'cog_renal_stage')): 0.9965230226516724,\n",
       " (('source', 'perineural_invasion'),\n",
       "  ('target', 'ovarian_surface_involvement')): 0.9963852763175964,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'selected_normal_targeted_sequencing')): 0.9963346719741821,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'extrascleral_extension')): 0.9963281154632568,\n",
       " (('source', 'tumor_Stage-Pathological'),\n",
       "  ('target', 'cog_liver_stage')): 0.9962989091873169,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'therapeutic_level_achieved')): 0.9961218237876892,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'kmer_content')): 0.996110200881958,\n",
       " (('source', 'tumor_laterality'),\n",
       "  ('target', 'biospecimen_laterality')): 0.996078372001648,\n",
       " (('source', 'tumor_stage'),\n",
       "  ('target', 'cog_liver_stage')): 0.9960553646087646,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'primary_disease')): 0.9960534572601318,\n",
       " (('source', 'tumor_stage_pathological'),\n",
       "  ('target', 'enneking_msts_stage')): 0.9959132075309753,\n",
       " (('source', 'CDKN2A.pathway.alteration'),\n",
       "  ('target', 'somatic_mutation_indexes')): 0.9958605766296387,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'methylation_beta_values')): 0.9951168298721313,\n",
       " (('source', 'CDKN2A.pathway.alteration'),\n",
       "  ('target', 'masked_somatic_mutations')): 0.9948688745498657,\n",
       " (('source', 'PN'), ('target', 'ajcc_pathologic_n')): 0.9943714141845703,\n",
       " (('source', 'PN'), ('target', 'uicc_pathologic_n')): 0.9941142797470093,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'rhabdoid_present')): 0.9940617084503174,\n",
       " (('source', 'FIGO_stage'), ('target', 'uicc_clinical_m')): 0.99354487657547,\n",
       " (('source', 'FIGO_stage'),\n",
       "  ('target', 'uicc_pathologic_m')): 0.9934516549110413,\n",
       " (('source', 'StromalScore'),\n",
       "  ('target', 'submitted_unaligned_reads_files')): 0.9931208491325378,\n",
       " (('source', 'ESTIMATEScore'),\n",
       "  ('target', 'submitted_unaligned_reads_files')): 0.9930949211120605,\n",
       " (('source', 'UMS'), ('target', 'msi_status')): 0.9930004477500916,\n",
       " (('source', 'tumor_laterality'), ('target', 'laterality')): 0.992915689945221,\n",
       " (('source', 'PN'), ('target', 'ajcc_clinical_n')): 0.9928981065750122,\n",
       " (('source', 'perineural_invasion'),\n",
       "  ('target', 'non_nodal_regional_disease')): 0.9919638633728027,\n",
       " (('source', 'CEA'), ('target', 'workflow_version')): 0.9918031692504883,\n",
       " (('source', 'CMS'),\n",
       "  ('target', 'somatic_aggregation_workflows')): 0.9916609525680542,\n",
       " (('source', 'PN'), ('target', 'uicc_clinical_n')): 0.9914900064468384,\n",
       " (('source', 'FIGO_stage'), ('target', 'ajcc_clinical_m')): 0.991349458694458,\n",
       " (('source', 'BMI'), ('target', 'average_base_quality')): 0.9909171462059021,\n",
       " (('source', 'ImmuneScore'),\n",
       "  ('target', 'submitted_unaligned_reads_files')): 0.9903726577758789,\n",
       " (('source', 'FIGO_stage'),\n",
       "  ('target', 'ajcc_pathologic_m')): 0.990330159664154,\n",
       " (('source', 'lymph_vascular_invasion'),\n",
       "  ('target', 'non_nodal_regional_disease')): 0.9902994632720947,\n",
       " (('source', 'BMI'), ('target', 'bmi')): 0.9899784922599792,\n",
       " (('source', 'tumor_site_curated'),\n",
       "  ('target', 'supratentorial_localization')): 0.9896013736724854,\n",
       " (('source', 'CMS'),\n",
       "  ('target', 'somatic_mutation_calling_workflows')): 0.989159345626831,\n",
       " (('source', 'histologic_type'),\n",
       "  ('target', 'pathology_details')): 0.9891501665115356,\n",
       " (('source', 'excluded_in_downstream_analysis'),\n",
       "  ('target', 'to_trim_adapter_sequence')): 0.9888603687286377,\n",
       " (('source', 'Stromal_fraction'),\n",
       "  ('target', 'hpv_positive_type')): 0.9887690544128418,\n",
       " (('source', 'tumor_stage_pathological'),\n",
       "  ('target', 'masaoka_stage')): 0.98838210105896,\n",
       " (('source', 'bmi'),\n",
       "  ('target', 'recist_targeted_regions_sum')): 0.9881073832511902,\n",
       " (('source', 'secondhand_smoke_exposure'),\n",
       "  ('target', 'cog_neuroblastoma_risk_group')): 0.9877168536186218,\n",
       " (('source', 'CMS'),\n",
       "  ('target', 'somatic_annotation_workflows')): 0.9875603914260864,\n",
       " (('source', 'secondhand_smoke_exposure'),\n",
       "  ('target', 'cog_rhabdomyosarcoma_risk_group')): 0.9875301122665405,\n",
       " (('source', 'histologic_type'),\n",
       "  ('target', 'pathology_reports')): 0.987454354763031,\n",
       " (('source', 'num_pack_years_sm'),\n",
       "  ('target', 'percent_lymphocyte_infiltration')): 0.9870725274085999,\n",
       " (('source', 'smoke_age_stop'),\n",
       "  ('target', 'hpv_positive_type')): 0.9860150218009949,\n",
       " (('source', 'number_of_lymph_nodes_positive_for_tumor'),\n",
       "  ('target', 'alcohol_drinks_per_day')): 0.9859881401062012,\n",
       " (('source', 'ESTIMATE.TumorPurity.rna'),\n",
       "  ('target', 'pmid')): 0.985005259513855,\n",
       " (('source', 'NAT_pathology_review'),\n",
       "  ('target', 'test_value_range')): 0.9847972393035889,\n",
       " (('source', 'TP53.mutation'),\n",
       "  ('target', 'analysis_metadata_files')): 0.9842020273208618,\n",
       " (('source', 'age'), ('target', 'percent_tumor_invasion')): 0.9839214086532593,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'rhabdoid_present')): 0.9833616018295288,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'rhabdoid_present')): 0.983079195022583,\n",
       " (('source', 'Case_ID'), ('target', 'bcr_id')): 0.9828305244445801,\n",
       " (('source', 'CD3.TILS.Counts'),\n",
       "  ('target', 'years_smoked')): 0.9826850891113281,\n",
       " (('source', 'Tumor_Size_cm'),\n",
       "  ('target', 'tumor_largest_dimension_diameter')): 0.9819190502166748,\n",
       " (('source', 'follow_up_period'),\n",
       "  ('target', 'exercise_frequency_weekly')): 0.9813328981399536,\n",
       " (('source', 'weight'), ('target', 'weight')): 0.9810733795166016,\n",
       " (('source', 'ethnicity'), ('target', 'ethnicity')): 0.9807141423225403,\n",
       " (('source', 'bmi'),\n",
       "  ('target', 'percent_neutrophil_infiltration')): 0.9802702069282532,\n",
       " (('source', 'Pack.Years.Smoked'),\n",
       "  ('target', 'percent_necrosis')): 0.9800798892974854,\n",
       " (('source', 'xCell.MicroenvironmentScore.rna'),\n",
       "  ('target', 'submitted_methylation_beta_values')): 0.9797173738479614,\n",
       " (('source', 'ER.Updated.Clinical.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.9794861078262329,\n",
       " (('source', 'patho_staging_pt'),\n",
       "  ('target', 'uicc_clinical_t')): 0.979246973991394,\n",
       " (('source', 'ERBB2.Updated.Clinical.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.9791682362556458,\n",
       " (('source', 'TNBC.Updated.Clinical.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.9790023565292358,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'micropapillary_features')): 0.9788367748260498,\n",
       " (('source', 'mutation_count'),\n",
       "  ('target', 'gross_tumor_weight')): 0.97855144739151,\n",
       " (('source', 'Tumor.Stage'),\n",
       "  ('target', 'ajcc_clinical_stage')): 0.9775514602661133,\n",
       " (('source', 'PR.Clinical.Status'),\n",
       "  ('target', 'peritoneal_washing_results')): 0.977135419845581,\n",
       " (('source', 'Grade'), ('target', 'tumor_grade')): 0.9765881896018982,\n",
       " (('source', 'country_of_origin'),\n",
       "  ('target', 'country_of_residence_at_enrollment')): 0.9764203429222107,\n",
       " (('source', 'ERBB2.Proteogenomic.Status'),\n",
       "  ('target', 'peritoneal_washing_results')): 0.9762623310089111,\n",
       " (('source', 'DNP.Count.wes'),\n",
       "  ('target', 'concentration')): 0.9761414527893066,\n",
       " (('source', 'case_id'),\n",
       "  ('target', 'distributor_reference')): 0.9754811525344849,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes-pN'),\n",
       "  ('target', 'target_capture_kit_name')): 0.9752646684646606,\n",
       " (('source', 'Participant'),\n",
       "  ('target', 'distributor_reference')): 0.975028395652771,\n",
       " (('source', 'neoAntigen_count'),\n",
       "  ('target', 'created_datetime')): 0.9739882349967957,\n",
       " (('source', 'TOP2A.Proteogenomic.Status'),\n",
       "  ('target', 'peritoneal_washing_results')): 0.973459780216217,\n",
       " (('source', 'KMT2C.Mutation.Type'),\n",
       "  ('target', 'experiment_metadata_files')): 0.9733126163482666,\n",
       " (('source', 'CD3_IHC_count'),\n",
       "  ('target', 'blood_test_normal_range_lower')): 0.9725768566131592,\n",
       " (('source', 'ESTIMATE.Stromal.Score'),\n",
       "  ('target', 'days_to_risk_factor')): 0.9723715782165527,\n",
       " (('source', 'bmi'), ('target', 'longest_dimension')): 0.9722659587860107,\n",
       " (('source', 'age'), ('target', 'initial_weight')): 0.971779465675354,\n",
       " (('source', 'xCell.StromaScore.rna'),\n",
       "  ('target', 'contamination')): 0.9716994762420654,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9714414477348328,\n",
       " (('source', 'vital_status'), ('target', 'vital_status')): 0.9713839292526245,\n",
       " (('source', 'bmi'), ('target', 'intermediate_dimension')): 0.9708595275878906,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'double_expressor_lymphoma')): 0.9706577658653259,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9705832004547119,\n",
       " (('source', 'PT'), ('target', 'ensat_pathologic_t')): 0.9704345464706421,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9702892303466797,\n",
       " (('source', 'smoking_inferred'),\n",
       "  ('target', 'evidence_of_progression_type')): 0.9700639247894287,\n",
       " (('source', 'Smoking.Score.Category.wes'),\n",
       "  ('target', 'evidence_of_progression_type')): 0.9700639247894287,\n",
       " (('source', 'neoAntigen_count'),\n",
       "  ('target', 'workflow_end_datetime')): 0.9698275327682495,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9694332480430603,\n",
       " (('source', 'Ischemia.Time.in.Minutes'),\n",
       "  ('target', 'imaging_suv_max')): 0.9689545631408691,\n",
       " (('source', 'Stage_mod'), ('target', 'iss_stage')): 0.9687494039535522,\n",
       " (('source', 'ER.Updated.Clinical.Status'),\n",
       "  ('target', 'imaging_result')): 0.9679863452911377,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9679648876190186,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'sequence_length_distribution')): 0.9672412872314453,\n",
       " (('source', 'cause_of_death'), ('target', 'comorbidity')): 0.9671681523323059,\n",
       " (('source', 'neoAntigen_count'),\n",
       "  ('target', 'updated_datetime')): 0.9670635461807251,\n",
       " (('source', 'ERBB2.Updated.Clinical.Status'),\n",
       "  ('target', 'imaging_result')): 0.9666072130203247,\n",
       " (('source', 'xCell.StromaScore.rna'),\n",
       "  ('target', 'contamination_error')): 0.9665648341178894,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_sequence_gc_content')): 0.9665376543998718,\n",
       " (('source', 'TNBC.Updated.Clinical.Status'),\n",
       "  ('target', 'imaging_result')): 0.966376781463623,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'therapeutic_levels_achieved')): 0.9661785364151001,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes-pN'),\n",
       "  ('target', 'target_capture_kit_version')): 0.966071367263794,\n",
       " (('source', 'neoAntigen_count'),\n",
       "  ('target', 'legacy_updated_datetime')): 0.9657502174377441,\n",
       " (('source', 'transcriptomic_subtype'),\n",
       "  ('target', 'peritoneal_fluid_cytological_status')): 0.9657330513000488,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes-pN'),\n",
       "  ('target', 'target_capture_kit_target_region')): 0.9653552770614624,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'transglottic_extension')): 0.9650874733924866,\n",
       " (('source', 'Muscle_fraction'),\n",
       "  ('target', 'staining_intensity_value')): 0.9650246500968933,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'alcohol_history')): 0.9647077322006226,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'radon_exposure')): 0.9646275043487549,\n",
       " (('source', 'tumor_site'),\n",
       "  ('target', 'lymph_node_dissection_site')): 0.9643396735191345,\n",
       " (('source', 'Fat_fraction'),\n",
       "  ('target', 'ecog_performance_status')): 0.9635006189346313,\n",
       " (('source', 'Ischemia.Time.in.Minutes'),\n",
       "  ('target', 'tumor_burden')): 0.9632821083068848,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'lymph_nodes_removed')): 0.9632496237754822,\n",
       " (('source', 'ProS'),\n",
       "  ('target', 'child_pugh_classification')): 0.9630151391029358,\n",
       " (('source', 'num_pack_years_sm'),\n",
       "  ('target', 'percent_granulocyte_infiltration')): 0.9630120396614075,\n",
       " (('source', 'expert_path_review_num_segments'),\n",
       "  ('target', 'imaging_date')): 0.9621822834014893,\n",
       " (('source', 'expert_path_review_num_segments'),\n",
       "  ('target', 'release_date')): 0.9617124795913696,\n",
       " (('source', 'xCell.StromaScore.rna'),\n",
       "  ('target', 'proportion_base_mismatch')): 0.9615234136581421,\n",
       " (('source', 'expert_path_review_num_segments'),\n",
       "  ('target', 'intended_release_date')): 0.9611648321151733,\n",
       " (('source', 'PIK3CA.pathway.alteration'),\n",
       "  ('target', 'experiment_name')): 0.9610876441001892,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'tumor_purity')): 0.9610277414321899,\n",
       " (('source', 'Histologic_Grade_FIGO'),\n",
       "  ('target', 'adverse_event_grade')): 0.9607592225074768,\n",
       " (('source', 'expert_path_review_num_segments'),\n",
       "  ('target', 'sequencing_date')): 0.9607036113739014,\n",
       " (('source', 'country'), ('target', 'country_of_birth')): 0.9606738090515137,\n",
       " (('source', 'Islet_fraction'),\n",
       "  ('target', 'peripancreatic_lymph_nodes_positive')): 0.9606590270996094,\n",
       " (('source', 'ProS'), ('target', 'analyte_type_id')): 0.96028733253479,\n",
       " (('source', 'CDKN2A.mutation'),\n",
       "  ('target', 'data_category')): 0.959856390953064,\n",
       " (('source', 'DNP.Count.GG.to.TT.or.CC.to.AA.wes'),\n",
       "  ('target', 'workflow_start_datetime')): 0.959845781326294,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'proportion_coverage_10x')): 0.9591566920280457,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'radon_exposure')): 0.958267092704773,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'alcohol_history')): 0.9580909013748169,\n",
       " (('source', 'Experiment'), ('target', 'margin_distance')): 0.9580385088920593,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'radon_exposure')): 0.9574429392814636,\n",
       " (('source', 'Path_Stage_Primary_Tumor-pT'),\n",
       "  ('target', 'stain_type')): 0.9573810696601868,\n",
       " (('source', 'Stage_mod'), ('target', 'clark_level')): 0.9573467969894409,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'alcohol_history')): 0.9572625756263733,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'methylation_beta_values')): 0.9572348594665527,\n",
       " (('source', 'MSH6_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.9568283557891846,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'lymph_nodes_removed')): 0.9558013677597046,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'lymph_nodes_removed')): 0.9549442529678345,\n",
       " (('source', 'Smoking.Signature.Fraction.wes'),\n",
       "  ('target', 'tobacco_smoking_quit_year')): 0.9537609815597534,\n",
       " (('source', 'P16'), ('target', 'test_result')): 0.9536643624305725,\n",
       " (('source', 'Non_neoplastic_duct'),\n",
       "  ('target', 'peripancreatic_lymph_nodes_positive')): 0.9524136781692505,\n",
       " (('source', 'tumor_site_original'),\n",
       "  ('target', 'metastasis_at_diagnosis_site')): 0.9523652791976929,\n",
       " (('source', 'chr_instability_idx'),\n",
       "  ('target', 'total_reads')): 0.9518246650695801,\n",
       " (('source', 'NRAS_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.951561689376831,\n",
       " (('source', 'race'), ('target', 'eye_color')): 0.951366662979126,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'relative_with_cancer_history')): 0.9506781101226807,\n",
       " (('source', 'medical_condition'),\n",
       "  ('target', 'family_histories')): 0.9503397941589355,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'relative_with_cancer_history')): 0.9500390291213989,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.9499878287315369,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'adapter_content')): 0.9496182203292847,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'sequence_duplication_levels')): 0.9485588073730469,\n",
       " (('source', 'Fat_fraction'), ('target', 'irs_stage')): 0.9484659433364868,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.9484125971794128,\n",
       " (('source', 'medical_condition'),\n",
       "  ('target', 'diagnoses')): 0.9482383131980896,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'proportion_coverage_30x')): 0.9480695128440857,\n",
       " (('source', 'BRCA2.mutation'),\n",
       "  ('target', 'secondary_expression_analyses')): 0.9480615854263306,\n",
       " (('source', 'Tumor_Size_cm'), ('target', 'test_value')): 0.9480257630348206,\n",
       " (('source', 'TP53.Mutation.Type'),\n",
       "  ('target', 'submitted_genomic_profiles')): 0.9479000568389893,\n",
       " (('source', 'Mucinous'), ('target', 'zygosity')): 0.9478833675384521,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9476750493049622,\n",
       " (('source', 'ESTIMATE_immune_score'),\n",
       "  ('target', 'days_to_lost_to_followup')): 0.9476234912872314,\n",
       " (('source', 'Clinical'), ('target', 'mutation_codon')): 0.9473625421524048,\n",
       " (('source', 'Tumor.Stage'), ('target', 'masaoka_stage')): 0.9470843076705933,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9468868374824524,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9465330839157104,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'coal_dust_exposure')): 0.946225643157959,\n",
       " (('source', 'age'),\n",
       "  ('target', 'epithelioid_cell_percent')): 0.9462059736251831,\n",
       " (('source', 'medical_condition'), ('target', 'cases')): 0.9460840225219727,\n",
       " (('source', 'PAM50'), ('target', 'test_analyte_type')): 0.9457953572273254,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9456371665000916,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9456216096878052,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'normal_tumor_genotype_snp_match')): 0.9455466270446777,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'coal_dust_exposure')): 0.9451910257339478,\n",
       " (('source', 'APOBEC.Signature'),\n",
       "  ('target', 'chromosome_arm')): 0.9449509382247925,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9445343017578125,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'gleason_score')): 0.9439481496810913,\n",
       " (('source', 'KMT2D.mutation'),\n",
       "  ('target', 'submitted_genomic_profiles')): 0.9436153173446655,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9434784054756165,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'coal_dust_exposure')): 0.9433398246765137,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'coal_dust_exposure')): 0.9428152441978455,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'bone_marrow_malignant_cells')): 0.9425647258758545,\n",
       " (('source', 'Tumor.Stage'),\n",
       "  ('target', 'uicc_clinical_stage')): 0.9423618316650391,\n",
       " (('source', 'Inflammation_fraction'),\n",
       "  ('target', 'gleason_grade_tertiary')): 0.942335307598114,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'size_extraocular_nodule')): 0.9422919750213623,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'coal_dust_exposure')): 0.9422850608825684,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'anaplasia_present')): 0.9420986771583557,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9416256546974182,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'alcohol_intensity')): 0.9414231181144714,\n",
       " (('source', 'TCGA pan-can immune clustering'),\n",
       "  ('target', 'drug_category')): 0.9410840272903442,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9410611391067505,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'alcohol_intensity')): 0.9410484433174133,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'relative_with_cancer_history')): 0.9408904910087585,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9405535459518433,\n",
       " (('source', 'CIMP.status.meth'),\n",
       "  ('target', 'disease_response')): 0.9402948617935181,\n",
       " (('source', 'ethnicity_self_identify'),\n",
       "  ('target', 'fragmentation_enzyme')): 0.9396735429763794,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'anaplasia_present')): 0.9396259784698486,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.9395641088485718,\n",
       " (('source', 'participant_country'),\n",
       "  ('target', 'country_of_residence_at_enrollment')): 0.9394738674163818,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'anaplasia_present')): 0.9392534494400024,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'anaplasia_present')): 0.9387047290802002,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'basic_statistics')): 0.9382449388504028,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.9381303191184998,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.9380857944488525,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'anaplasia_present')): 0.9380618929862976,\n",
       " (('source', 'Tumor.Status'),\n",
       "  ('target', 'classification')): 0.9380562901496887,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'coal_dust_exposure')): 0.9379018545150757,\n",
       " (('source', 'weight'),\n",
       "  ('target', 'fev1_fvc_post_bronch_percent')): 0.9378745555877686,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'anaplasia_present')): 0.9372946619987488,\n",
       " (('source', 'Smoking.History_modified'),\n",
       "  ('target', 'type_of_tobacco_used')): 0.9371801614761353,\n",
       " (('source', 'ESTIMATE_stromal_score'),\n",
       "  ('target', 'age_at_last_exposure')): 0.9371542930603027,\n",
       " (('source', 'Inflammation_fraction'),\n",
       "  ('target', 'primary_gleason_grade')): 0.9367823600769043,\n",
       " (('source', 'ERBB2.Updated.Clinical.Status'),\n",
       "  ('target', 'margin_status')): 0.9367731809616089,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.9367102384567261,\n",
       " (('source', 'Subsite'),\n",
       "  ('target', 'biospecimen_anatomic_site')): 0.9366335868835449,\n",
       " (('source', 'TSNet.Purity.rna'),\n",
       "  ('target', 'total_sequences')): 0.9366182088851929,\n",
       " (('source', 'TNBC.Updated.Clinical.Status'),\n",
       "  ('target', 'margin_status')): 0.9365473985671997,\n",
       " (('source', 'Smoking.Signature.Fraction.wes'),\n",
       "  ('target', 'pancreatitis_onset_year')): 0.9365221261978149,\n",
       " (('source', 'NOTCH1.mutation'),\n",
       "  ('target', 'submitted_genomic_profiles')): 0.9364538192749023,\n",
       " (('source', 'ESTIMATE.Immune.Score'),\n",
       "  ('target', 'fragment_maximum_length')): 0.9362831115722656,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'perineural_invasion_present')): 0.9360915422439575,\n",
       " (('source', 'ER.Updated.Clinical.Status'),\n",
       "  ('target', 'margin_status')): 0.935616135597229,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'proportion_reads_mapped')): 0.9355520606040955,\n",
       " (('source', 'ESTIMATE.Immune.Score'),\n",
       "  ('target', 'fragment_mean_length')): 0.9353929162025452,\n",
       " (('source', 'ESTIMATE.Immune.Score'),\n",
       "  ('target', 'average_insert_size')): 0.9352390766143799,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'gleason_patterns_percent')): 0.9344609975814819,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.9343390464782715,\n",
       " (('source', 'ESTIMATE.StromalScore.rna'),\n",
       "  ('target', 'days_to_birth')): 0.9337878227233887,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.9336715340614319,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'per_base_n_content')): 0.9335723519325256,\n",
       " (('source', 'CIBERSORT.Absolute.Score'),\n",
       "  ('target', 'body_surface_area')): 0.9335368871688843,\n",
       " (('source', 'number_of_lymph_nodes_examined'),\n",
       "  ('target', 'number_of_fractions')): 0.9327996969223022,\n",
       " (('source', 'CD3.TILS.Counts'),\n",
       "  ('target', 'karnofsky_performance_status')): 0.9322673082351685,\n",
       " (('source', 'number_of_lymph_nodes_positive_for_tumor'),\n",
       "  ('target', 'exposure_duration_hrs_per_day')): 0.9322460889816284,\n",
       " (('source', 'Chromosome.INstability.index.CIN.'),\n",
       "  ('target', 'number_proliferating_cells')): 0.9322149157524109,\n",
       " (('source', 'Pathology.based.histology.assessment'),\n",
       "  ('target', 'center_type')): 0.931953489780426,\n",
       " (('source', 'mutation_rate'), ('target', 'transcript')): 0.9315310716629028,\n",
       " (('source', 'Inflammation_fraction'),\n",
       "  ('target', 'secondary_gleason_grade')): 0.9313105344772339,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'sample_type_id')): 0.9310970306396484,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'sample_ordinal')): 0.9309695959091187,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'environmental_tobacco_smoke_exposure')): 0.930944561958313,\n",
       " (('source', 'xCell.ImmuneScore.rna'),\n",
       "  ('target', 'submitted_methylation_beta_values')): 0.9302084445953369,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'breslow_thickness')): 0.9296680092811584,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'lane_number')): 0.929628849029541,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'analyte_quantity')): 0.9296189546585083,\n",
       " (('source', 'Stage'), ('target', 'irs_group')): 0.9286117553710938,\n",
       " (('source', 'ESTIMATE_stromal_score'),\n",
       "  ('target', 'prostatic_chips_total_count')): 0.9285545349121094,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'year_of_follow_up')): 0.9283636212348938,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'proportion_coverage_10x')): 0.9282727241516113,\n",
       " (('source', 'pathologic_staging_primary_tumor_pt'),\n",
       "  ('target', 'myasthenia_gravis_classification')): 0.9279483556747437,\n",
       " (('source', 'CUL3.mutation'), ('target', 'error_type')): 0.9268709421157837,\n",
       " (('source', 'Ethnicity_Self_Identify'),\n",
       "  ('target', 'demographics')): 0.9265405535697937,\n",
       " (('source', 'Mutation'),\n",
       "  ('target', 'tumor_grade_category')): 0.9255441427230835,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'melanoma_known_primary')): 0.9249367117881775,\n",
       " (('source', 'pathologic_staging_distant_metastasis_pm'),\n",
       "  ('target', 'residual_tumor')): 0.9249271154403687,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'per_base_sequence_content')): 0.924510657787323,\n",
       " (('source', 'Mutation.Count.ExcludingINDELs.wes'),\n",
       "  ('target', 'tumor_depth')): 0.924363374710083,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'recist_targeted_regions_number')): 0.9238704442977905,\n",
       " (('source', 'mutation_count'),\n",
       "  ('target', 'blood_test_normal_range_upper')): 0.9229249954223633,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_tile_sequence_quality')): 0.9228055477142334,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'sites_of_involvement_count')): 0.9225094318389893,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'treatment_or_therapy')): 0.9221473932266235,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'tumor_ploidy')): 0.9217033386230469,\n",
       " (('source', 'Type'), ('target', 'treatments')): 0.921698808670044,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'major_version')): 0.9216546416282654,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'lost_to_followup')): 0.9215330481529236,\n",
       " (('source', 'Cigarettes.per.Day'),\n",
       "  ('target', 'time_between_waking_and_first_smoke')): 0.9207443594932556,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'treatment_or_therapy')): 0.9206907749176025,\n",
       " (('source', 'mRNA-seq'),\n",
       "  ('target', 'tumor_grade_category')): 0.920560359954834,\n",
       " (('source', 'Islet_fraction'),\n",
       "  ('target', 'staining_intensity_value')): 0.9204078912734985,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'lost_to_followup')): 0.920039713382721,\n",
       " (('source', 'Muscle_fraction'),\n",
       "  ('target', 'peripancreatic_lymph_nodes_positive')): 0.9197812080383301,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'treatment_or_therapy')): 0.9193031191825867,\n",
       " (('source', 'TMT Proteome (Normal)'),\n",
       "  ('target', 'prescribed_dose')): 0.9192618131637573,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'lost_to_followup')): 0.9187703132629395,\n",
       " (('source', 'CIMP.status.meth'),\n",
       "  ('target', 'best_overall_response')): 0.9186772108078003,\n",
       " (('source', 'TMT Phosphoproteome (Normal)'),\n",
       "  ('target', 'prescribed_dose')): 0.9184728860855103,\n",
       " (('source', 'race'), ('target', 'channel')): 0.9182276725769043,\n",
       " (('source', 'additional_pathologic_findings'),\n",
       "  ('target', 'additional_pathology_findings')): 0.9180496335029602,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'treatment_or_therapy')): 0.9178599119186401,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'treatment_or_therapy')): 0.917316198348999,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'lost_to_followup')): 0.9172998666763306,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'course_number')): 0.9168961048126221,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'lost_to_followup')): 0.9165667295455933,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9157766103744507,\n",
       " (('source', 'MAP3K1.Mutation.Type'),\n",
       "  ('target', 'file_name')): 0.915513277053833,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'lymph_nodes_positive')): 0.9154866933822632,\n",
       " (('source', 'Histologic_Type'),\n",
       "  ('target', 'procedures_performed')): 0.9150265455245972,\n",
       " (('source', 'Smoking.score.wes'),\n",
       "  ('target', 'filtered_copy_number_segments')): 0.9149903059005737,\n",
       " (('source', 'cause_of_death'), ('target', 'comorbidities')): 0.91487717628479,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9146683216094971,\n",
       " (('source', 'Channel'), ('target', 'aa_change')): 0.9143906235694885,\n",
       " (('source', 'Sample.ID'),\n",
       "  ('target', 'multiplex_barcode')): 0.9125536680221558,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9124446511268616,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9120380878448486,\n",
       " (('source', 'NMF.subtype.core'),\n",
       "  ('target', 'molecular_consequence')): 0.9119876623153687,\n",
       " (('source', 'NMF.subtype'),\n",
       "  ('target', 'molecular_consequence')): 0.9117234945297241,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'treatment_or_therapy')): 0.9116306304931641,\n",
       " (('source', 'Tumor.Stage'),\n",
       "  ('target', 'enneking_msts_stage')): 0.9113366603851318,\n",
       " (('source', 'Smoking.score.wes'), ('target', 'revision')): 0.9112821817398071,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.9111536741256714,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'lost_to_followup')): 0.9109582304954529,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9108819961547852,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.9108716249465942,\n",
       " (('source', 'Path_Stage_Dist_Mets-pM'),\n",
       "  ('target', 'treatment_effect')): 0.9103347063064575,\n",
       " (('source', 'TCGA.RNA.cluster'),\n",
       "  ('target', 'variant_origin')): 0.9101599454879761,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'pregnancy_count')): 0.909947395324707,\n",
       " (('source', 'tobacco_smoking_history'),\n",
       "  ('target', 'alcohol_frequency')): 0.90969318151474,\n",
       " (('source', 'expert_path_review_details'),\n",
       "  ('target', 'target_capture_kit')): 0.9094310998916626,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.9094287157058716,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'treatment_dose_max')): 0.9093488454818726,\n",
       " (('source', 'smoking_history'),\n",
       "  ('target', 'alcohol_frequency')): 0.909023642539978,\n",
       " (('source', 'number_of_lymph_nodes_positive_for_tumor'),\n",
       "  ('target', 'hiv_viral_load')): 0.9089788198471069,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'copy_number')): 0.9078538417816162,\n",
       " (('source', 'MSmutectFishercall'),\n",
       "  ('target', 'library_strategy')): 0.9077041149139404,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'prostatic_involvement_percent')): 0.9069039821624756,\n",
       " (('source', 'Smoking.History'),\n",
       "  ('target', 'alcohol_frequency')): 0.9065349698066711,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'radiosensitizing_agent')): 0.9059683084487915,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'analyte_volume')): 0.9056925773620605,\n",
       " (('source', 'Age'), ('target', 'weiss_assessment_score')): 0.9051082134246826,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'copy_number_estimates')): 0.9047200083732605,\n",
       " (('source', 'GATA3.Mutation.Type'),\n",
       "  ('target', 'file_state')): 0.9047194123268127,\n",
       " (('source', 'lymph_vascular_invasion'),\n",
       "  ('target', 'micropapillary_features')): 0.9038896560668945,\n",
       " (('source', 'ESTIMATE.TumorPurity'),\n",
       "  ('target', 'copy_number_segments')): 0.9035123586654663,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.9031849503517151,\n",
       " (('source', 'Hypermutation_status'),\n",
       "  ('target', 'preservation_method')): 0.9031664133071899,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'discontiguous_lesion_count')): 0.9025470018386841,\n",
       " (('source', 'Age.in.Month'),\n",
       "  ('target', 'weiss_assessment_score')): 0.9024488925933838,\n",
       " (('source', 'Chromosome.INstability.index.CIN.'),\n",
       "  ('target', 'age_at_diagnosis')): 0.9017720222473145,\n",
       " (('source', 'Race'), ('target', 'eye_color')): 0.9007326364517212,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'aliquot_quantity')): 0.9005612134933472,\n",
       " (('source', 'Acinar_fraction'),\n",
       "  ('target', 'chromosomal_translocation')): 0.9004001617431641,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'melanoma_known_primary')): 0.9003276824951172,\n",
       " (('source', 'Non_neoplastic_duct'),\n",
       "  ('target', 'staining_intensity_value')): 0.9001999497413635,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'aliquot_volume')): 0.8999468088150024,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'melanoma_known_primary')): 0.8993365168571472,\n",
       " (('source', 'Age'), ('target', 'tumor_code_id')): 0.8986877799034119,\n",
       " (('source', 'Total.Mutation.Count.per.Mb.wes'),\n",
       "  ('target', 'rna_integrity_number')): 0.8986514806747437,\n",
       " (('source', 'tumor_stage_pathological'),\n",
       "  ('target', 'ajcc_pathologic_stage')): 0.8985543251037598,\n",
       " (('source', 'Tumor_Site_Other'),\n",
       "  ('target', 'site_of_resection_or_biopsy')): 0.8980996608734131,\n",
       " (('source', 'Type'),\n",
       "  ('target', 'regimen_or_line_of_therapy')): 0.8979072570800781,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'tumor_purity')): 0.897834837436676,\n",
       " (('source', 'Tumor.Stage'),\n",
       "  ('target', 'ajcc_pathologic_stage')): 0.8978028297424316,\n",
       " (('source', 'AKT1.Mutation.Type'),\n",
       "  ('target', 'simple_germline_variations')): 0.8974730968475342,\n",
       " (('source', 'NFE2L2.mutation'),\n",
       "  ('target', 'simple_germline_variations')): 0.8974730968475342,\n",
       " (('source', 'KEAP1.mutation'),\n",
       "  ('target', 'simple_germline_variations')): 0.8974730968475342,\n",
       " (('source', 'PIK3CA.mutation'),\n",
       "  ('target', 'simple_germline_variations')): 0.8974729776382446,\n",
       " (('source', 'ESTIMATE.Immune.Score'),\n",
       "  ('target', 'fragment_minimum_length')): 0.8969426155090332,\n",
       " (('source', 'Polyps_Present'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.8969234228134155,\n",
       " (('source', 'Cigarettes.per.Day'),\n",
       "  ('target', 'karnofsky_performance_status')): 0.8967675566673279,\n",
       " (('source', 'CIN.wes'), ('target', 'mitotic_total_area')): 0.8964991569519043,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'undescended_testis_corrected')): 0.8963555693626404,\n",
       " (('source', 'Tumor_Site'),\n",
       "  ('target', 'tissue_or_organ_of_origin')): 0.8961855173110962,\n",
       " (('source', 'PIK3CA.Mutation.Type'),\n",
       "  ('target', 'aligned_reads_files')): 0.8956526517868042,\n",
       " (('source', 'Vascular_Invasion'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.8955304622650146,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'proportion_base_mismatch')): 0.8950867652893066,\n",
       " (('source', 'Polyps_History'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.8950778245925903,\n",
       " (('source', 'ImmuneCluster.rna'),\n",
       "  ('target', 'composition')): 0.8946558237075806,\n",
       " (('source', 'perineural_invasion'),\n",
       "  ('target', 'micropapillary_features')): 0.8940346240997314,\n",
       " (('source', 'Synchronous_Tumors'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.8937743306159973,\n",
       " (('source', 'ARID1A.Mutation.Type'),\n",
       "  ('target', 'submitted_aligned_reads_files')): 0.8936183452606201,\n",
       " (('source', 'TSNet.Purity.rna'), ('target', 'file_size')): 0.8935723304748535,\n",
       " (('source', 'miRNA-seq'),\n",
       "  ('target', 'dlco_ref_predictive_percent')): 0.8933343887329102,\n",
       " (('source', 'Histologic_type'), ('target', 'roots')): 0.8929511308670044,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_sequence_quality_score')): 0.892875075340271,\n",
       " (('source', 'xCell.Stromal.Score'),\n",
       "  ('target', 'days_to_imaging')): 0.892621636390686,\n",
       " (('source', 'CBFB.Mutation.Type'),\n",
       "  ('target', 'archives')): 0.8920835256576538,\n",
       " (('source', 'Secondhand.Smoke'),\n",
       "  ('target', 'cause_of_death')): 0.8918966054916382,\n",
       " (('source', 'path_diag_to_last_contact_days'),\n",
       "  ('target', 'relatives_with_cancer_history_count')): 0.8917937874794006,\n",
       " (('source', 'lymph_vascular_invasion'),\n",
       "  ('target', 'transglottic_extension')): 0.8908960223197937,\n",
       " (('source', 'Lymphatic_Invasion'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.8904043436050415,\n",
       " (('source', 'residual_tumor'),\n",
       "  ('target', 'lymph_node_dissection_method')): 0.8902575969696045,\n",
       " (('source', 'expert_path_review_details'),\n",
       "  ('target', 'project_id')): 0.8898577094078064,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'ulceration_indicator')): 0.8895097970962524,\n",
       " (('source', 'PAM50'), ('target', 'analyte_type')): 0.889098048210144,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'contamination')): 0.8889827728271484,\n",
       " (('source', 'patho_staging_pn'),\n",
       "  ('target', 'second_gene_symbol')): 0.8885444402694702,\n",
       " (('source', 'Histologic_type'),\n",
       "  ('target', 'hysterectomy_margins_involved')): 0.8880319595336914,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'minor_version')): 0.8879216909408569,\n",
       " (('source', 'MLLT4.Mutation.Type'),\n",
       "  ('target', 'copy_number_liftover_workflows')): 0.8877705931663513,\n",
       " (('source', 'NFE2L2.mutation'),\n",
       "  ('target', 'library_strand')): 0.8877567052841187,\n",
       " (('source', 'AKT1.Mutation.Type'),\n",
       "  ('target', 'library_strand')): 0.8877565860748291,\n",
       " (('source', 'KEAP1.mutation'),\n",
       "  ('target', 'library_strand')): 0.8877565860748291,\n",
       " (('source', 'PIK3CA.mutation'),\n",
       "  ('target', 'library_strand')): 0.8877565860748291,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'synchronous_malignancy')): 0.8868973851203918,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'synchronous_malignancy')): 0.8863144516944885,\n",
       " (('source', 'Path_Stage_Primary_Tumor_pT'),\n",
       "  ('target', 'ajcc_clinical_t')): 0.8861995935440063,\n",
       " (('source', 'Perineural_Invasion'),\n",
       "  ('target', 'columnar_mucosa_present')): 0.886157214641571,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'ulceration_indicator')): 0.8860427737236023,\n",
       " (('source', 'Sample.ID'), ('target', 'chipseq_target')): 0.8858749866485596,\n",
       " (('source', 'chr_instability_idx'),\n",
       "  ('target', 'cell_count')): 0.8858577013015747,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'proportion_coverage_30x')): 0.8858243227005005,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'age_at_onset')): 0.8858121633529663,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'synchronous_malignancy')): 0.8857619762420654,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'ulceration_indicator')): 0.8855129480361938,\n",
       " (('source', 'smoke_age_start'), ('target', 'revision')): 0.8850705623626709,\n",
       " (('source', 'Tumor_Site_Other'),\n",
       "  ('target', 'progression_or_recurrence_anatomic_site')): 0.8846026062965393,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'ovarian_surface_involvement')): 0.8841740489006042,\n",
       " (('source', 'smoking_second_hand'),\n",
       "  ('target', 'cause_of_death')): 0.8840039968490601,\n",
       " (('source', 'Secondhand.Smoke'),\n",
       "  ('target', 'initial_disease_status')): 0.8839054107666016,\n",
       " (('source', 'Pathology.based.histology.assessment'),\n",
       "  ('target', 'centers')): 0.8835939764976501,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'overrepresented_sequences')): 0.8826464414596558,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'parent_with_radiation_exposure')): 0.8824422359466553,\n",
       " (('source', 'tumor_site_original'),\n",
       "  ('target', 'sites_of_involvement')): 0.8824187517166138,\n",
       " (('source', 'bmi'), ('target', 'necrosis_percent')): 0.8823549747467041,\n",
       " (('source', 'exclusion_reason'),\n",
       "  ('target', 'other_clinical_attributes')): 0.8823239803314209,\n",
       " (('source', 'CIBERSORT.AbsoluteScore.rna'),\n",
       "  ('target', 'days_to_last_follow_up')): 0.8817561864852905,\n",
       " (('source', 'smoking_second_hand'),\n",
       "  ('target', 'initial_disease_status')): 0.8815585374832153,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'double_hit_lymphoma')): 0.8813822865486145,\n",
       " (('source', 'perineural_invasion'),\n",
       "  ('target', 'transglottic_extension')): 0.8811501264572144,\n",
       " (('source', 'normal_included_for_the_study'),\n",
       "  ('target', 'consistent_pathology_review')): 0.8811448216438293,\n",
       " (('source', 'tumor_site_curation_notes'),\n",
       "  ('target', 'comorbidity_method_of_diagnosis')): 0.8801538944244385,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'tumor_width_measurement')): 0.8798837661743164,\n",
       " (('source', 'additional_pathologic_findings'),\n",
       "  ('target', 'primary_diagnosis')): 0.8794701099395752,\n",
       " (('source', 'SampleID'), ('target', 'well_number')): 0.8793912529945374,\n",
       " (('source', 'SOX2.cna.wes'), ('target', 'schema_version')): 0.879075288772583,\n",
       " (('source', 'PT'), ('target', 'uicc_pathologic_t')): 0.8789564371109009,\n",
       " (('source', 'TOP2A.Proteogenomic.Status'),\n",
       "  ('target', 'imaging_result')): 0.8787718415260315,\n",
       " (('source', 'transcriptomic_subtype'),\n",
       "  ('target', 'lymph_node_involved_site')): 0.8783819675445557,\n",
       " (('source', 'smoking_history'),\n",
       "  ('target', 'cdc_hiv_risk_factors')): 0.8778617978096008,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'days_to_consent')): 0.8776125907897949,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'days_to_consent')): 0.8773490190505981,\n",
       " (('source', 'tobacco_smoking_history'),\n",
       "  ('target', 'cdc_hiv_risk_factors')): 0.8771665096282959,\n",
       " (('source', 'num_pack_years_sm'),\n",
       "  ('target', 'fragment_standard_deviation_length')): 0.8771536350250244,\n",
       " (('source', 'Vital.Status'),\n",
       "  ('target', 'relationship_type')): 0.8768921494483948,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'days_to_consent')): 0.876282274723053,\n",
       " (('source', 'stemness_score'), ('target', 'height')): 0.8760857582092285,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'course_number')): 0.8757237195968628,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'pregnant_at_diagnosis')): 0.8755686283111572,\n",
       " (('source', 'CIBERSORT.Absolute.Score'),\n",
       "  ('target', 'a260_a280_ratio')): 0.8754379153251648,\n",
       " (('source', 'xCell.Stromal.Score'),\n",
       "  ('target', 'days_to_collection')): 0.8752866387367249,\n",
       " (('source', 'Smoking.History'),\n",
       "  ('target', 'cdc_hiv_risk_factors')): 0.8752577304840088,\n",
       " (('source', 'cause_of_death'), ('target', 'risk_factor')): 0.8749417066574097,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'melanoma_known_primary')): 0.8741805553436279,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'days_to_consent')): 0.8739842772483826,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'oxygen_use_indicator')): 0.8738065958023071,\n",
       " (('source', 'Histologic_Type'),\n",
       "  ('target', 'method_of_diagnosis')): 0.873792290687561,\n",
       " (('source', 'Mutation.Count.ExcludingINDELs.wes'),\n",
       "  ('target', 'percent_eosinophil_infiltration')): 0.8727892637252808,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_base_sequence_quality')): 0.8718750476837158,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'proportion_reads_duplicated')): 0.8715115785598755,\n",
       " (('source', 'exclusion_reason'),\n",
       "  ('target', 'laboratory_test')): 0.8706358671188354,\n",
       " (('source', 'cause_of_death'),\n",
       "  ('target', 'adverse_event')): 0.870489776134491,\n",
       " (('source', 'Total.Mutation.Count.wes'),\n",
       "  ('target', 'tumor_depth')): 0.8694351315498352,\n",
       " (('source', 'StromalScore'), ('target', 'doi')): 0.8691911697387695,\n",
       " (('source', 'ESTIMATEScore'), ('target', 'doi')): 0.8686707019805908,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'recist_targeted_regions_number')): 0.8684202432632446,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'sites_of_involvement_count')): 0.8683727383613586,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'tumor_ploidy')): 0.8681328296661377,\n",
       " (('source', 'SampleID'), ('target', 'plate_well')): 0.8678818941116333,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'proportion_reads_mapped')): 0.8674618005752563,\n",
       " (('source', 'ERBB2.Proteogenomic.Status'),\n",
       "  ('target', 'imaging_result')): 0.8672824501991272,\n",
       " (('source', 'TCGA pan-can immune clustering'),\n",
       "  ('target', 'nononcologic_therapeutic_agents')): 0.8671308755874634,\n",
       " (('source', 'TumorPurity'),\n",
       "  ('target', 'flow_cell_barcode')): 0.8668375015258789,\n",
       " (('source', 'CDKN2A.cna.wes'), ('target', 'locus')): 0.8667086362838745,\n",
       " (('source', 'ImmuneCluster.rna'), ('target', 'aliquots')): 0.8663477301597595,\n",
       " (('source', 'Channel'),\n",
       "  ('target', 'tissue_microarray_coordinates')): 0.8660145998001099,\n",
       " (('source', 'is_this_patient_lost_to_follow_up'),\n",
       "  ('target', 'parent_with_radiation_exposure')): 0.8658726215362549,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'filtered_copy_number_segments')): 0.8652443885803223,\n",
       " (('source', 'MLLT4.Mutation.Type'), ('target', 'name')): 0.865200400352478,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'immunosuppressive_treatment_type')): 0.8650433421134949,\n",
       " (('source', 'Channel'), ('target', 'chip_position')): 0.8648624420166016,\n",
       " (('source', 'ImmuneScore'), ('target', 'doi')): 0.8647762537002563,\n",
       " (('source', 'tumor_pathology_review'),\n",
       "  ('target', 'clinical_supplements')): 0.8647483587265015,\n",
       " (('source', 'lost_to_follow_up'),\n",
       "  ('target', 'parent_with_radiation_exposure')): 0.8646820783615112,\n",
       " (('source', 'age'),\n",
       "  ('target', 'fev1_fvc_pre_bronch_percent')): 0.8643238544464111,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'major_version')): 0.8641947507858276,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'immunosuppressive_treatment_type')): 0.8638221621513367,\n",
       " (('source', 'Sample.IDs'), ('target', 'exon')): 0.8637712001800537,\n",
       " (('source', 'Smoking.History_modified'),\n",
       "  ('target', 'type_of_smoke_exposure')): 0.8632668256759644,\n",
       " (('source', 'clinical_staging_distant_metastasis_cm'),\n",
       "  ('target', 'data_format')): 0.8629699945449829,\n",
       " (('source', 'PR.Clinical.Status'),\n",
       "  ('target', 'imaging_result')): 0.8623383045196533,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'fragment_minimum_length')): 0.862018346786499,\n",
       " (('source', 'TMT.Plex'),\n",
       "  ('target', 'time_between_clamping_and_freezing')): 0.861216127872467,\n",
       " (('source', 'SOX2.cna.wes'),\n",
       "  ('target', 'base_caller_version')): 0.861161470413208,\n",
       " (('source', 'mRNA-seq'),\n",
       "  ('target', 'tumor_infiltrating_macrophages')): 0.8602190613746643,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'cd4_count')): 0.8599521517753601,\n",
       " (('source', 'follow_up_days'),\n",
       "  ('target', 'lymph_nodes_positive')): 0.8599306344985962,\n",
       " (('source', 'tumor_stage_pathological'),\n",
       "  ('target', 'ajcc_clinical_stage')): 0.8595738410949707,\n",
       " (('source', 'smoking_inferred'),\n",
       "  ('target', 'evidence_of_recurrence_type')): 0.8591873049736023,\n",
       " (('source', 'Smoking.Score.Category.wes'),\n",
       "  ('target', 'evidence_of_recurrence_type')): 0.8591873049736023,\n",
       " (('source', 'Mutation'),\n",
       "  ('target', 'tumor_infiltrating_macrophages')): 0.8591232299804688,\n",
       " (('source', 'pathologic_staging_primary_tumor_pt'),\n",
       "  ('target', 'papillary_renal_cell_type')): 0.8591181039810181,\n",
       " (('source', 'Tumor_Site'),\n",
       "  ('target', 'tumor_level_prostate')): 0.8590046167373657,\n",
       " (('source', 'TumorPurity'), ('target', 'treatment_arm')): 0.8589186072349548,\n",
       " (('source', 'smoke_age_start'),\n",
       "  ('target', 'sample_ordinal')): 0.8586610555648804,\n",
       " (('source', 'Number.of.non.synonymous.Mutations'),\n",
       "  ('target', 'days_to_treatment_start')): 0.8581405878067017,\n",
       " (('source', 'follow_up_days'),\n",
       "  ('target', 'discontiguous_lesion_count')): 0.8578919172286987,\n",
       " (('source', 'ESTIMATE_immune_score'),\n",
       "  ('target', 'read_length')): 0.8577651381492615,\n",
       " (('source', 'CIBERSORT.AbsoluteScore.rna'),\n",
       "  ('target', 'age_at_diagnosis')): 0.8572759628295898,\n",
       " (('source', 'tumor_site_curation_notes'),\n",
       "  ('target', 'risk_factor_method_of_diagnosis')): 0.8569930791854858,\n",
       " (('source', 'NF1.mutation'), ('target', 'file_name')): 0.8569838404655457,\n",
       " (('source', 'Race'), ('target', 'channel')): 0.856899082660675,\n",
       " (('source', 'clinical_staging_distant_metastasis_cm'),\n",
       "  ('target', 'short_name')): 0.8566081523895264,\n",
       " (('source', 'SampleID'),\n",
       "  ('target', 'figo_staging_edition_year')): 0.8564385175704956,\n",
       " (('source', 'Mucinous'), ('target', 'aids_risk_factors')): 0.8562142848968506,\n",
       " (('source', 'histology_diagnosis'),\n",
       "  ('target', 'exposures')): 0.8558076620101929,\n",
       " (('source', 'TMT.Plex'),\n",
       "  ('target', 'occupation_duration_years')): 0.85560142993927,\n",
       " (('source', 'Sample.ID'), ('target', 'chromosome')): 0.8553764820098877,\n",
       " (('source', 'Smoking.History'),\n",
       "  ('target', 'hormonal_contraceptive_use')): 0.8553553819656372,\n",
       " (('source', 'mRNA-seq'),\n",
       "  ('target', 'tumor_infiltrating_lymphocytes')): 0.8552370071411133,\n",
       " (('source', 'follow_up_period'),\n",
       "  ('target', 'smoking_frequency')): 0.8551326990127563,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'days_to_first_event')): 0.8550097942352295,\n",
       " (('source', 'Subsite'),\n",
       "  ('target', 'contiguous_organ_invaded')): 0.8547153472900391,\n",
       " (('source', 'Mutation'),\n",
       "  ('target', 'tumor_infiltrating_lymphocytes')): 0.854145884513855,\n",
       " (('source', 'Histologic_Grade_FIGO'),\n",
       "  ('target', 'staining_intensity_scale')): 0.8540553450584412,\n",
       " (('source', 'patho_staging_pt'),\n",
       "  ('target', 'uicc_pathologic_t')): 0.8533533811569214,\n",
       " (('source', 'histology_diagnosis'),\n",
       "  ('target', 'history_of_tumor_type')): 0.8532358407974243,\n",
       " (('source', 'CIN.wes'), ('target', 'cell_count')): 0.8532077074050903,\n",
       " (('source', 'exclusion_reason'),\n",
       "  ('target', 'cause_of_response')): 0.8531410694122314,\n",
       " (('source', 'follow_up_days'), ('target', 'copy_number')): 0.8528258800506592,\n",
       " (('source', 'Chromosome.INstability.index.CIN.'),\n",
       "  ('target', 'days_to_follow_up')): 0.8518530130386353,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes_pN'),\n",
       "  ('target', 'platforms')): 0.8517578840255737,\n",
       " (('source', 'tumor_site'),\n",
       "  ('target', 'treatment_anatomic_sites')): 0.8515342473983765,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'proportion_targets_no_coverage')): 0.8514300584793091,\n",
       " (('source', 'Tumor_Size_cm'),\n",
       "  ('target', 'tumor_length_measurement')): 0.8507654666900635,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'enneking_msts_tumor_site')): 0.8505079746246338,\n",
       " (('source', 'smoking_history'),\n",
       "  ('target', 'hormonal_contraceptive_use')): 0.8494390845298767,\n",
       " (('source', 'MAP3K1.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.849430501461029,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes_pN'),\n",
       "  ('target', 'instrument_model')): 0.8494228720664978,\n",
       " (('source', 'P16'), ('target', 'igcccg_stage')): 0.8494085669517517,\n",
       " (('source', 'BMI'),\n",
       "  ('target', 'percent_neutrophil_infiltration')): 0.8491390943527222,\n",
       " (('source', 'SF3B1.Mutation.Type'),\n",
       "  ('target', 'fastq_name')): 0.8490053415298462,\n",
       " (('source', 'KMT2C.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8489259481430054,\n",
       " (('source', 'PIK3CA.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8489258885383606,\n",
       " (('source', 'NFE2L2.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8489258289337158,\n",
       " (('source', 'tobacco_smoking_history'),\n",
       "  ('target', 'hormonal_contraceptive_use')): 0.8487091064453125,\n",
       " (('source', 'bmi'), ('target', 'average_base_quality')): 0.8483449220657349,\n",
       " (('source', 'xCell.Stromal.Score'),\n",
       "  ('target', 'days_to_adverse_event')): 0.8481366634368896,\n",
       " (('source', 'ARID1A.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8479446172714233,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'tumor_thickness')): 0.847848653793335,\n",
       " (('source', 'GATA3.Mutation.Type'),\n",
       "  ('target', 'copy_number_auxiliary_files')): 0.8477169275283813,\n",
       " (('source', 'TCGA pan-can immune clustering'),\n",
       "  ('target', 'biospecimen_supplements')): 0.8474019765853882,\n",
       " (('source', 'Mutation.Count.Excluding.Silent.wes'),\n",
       "  ('target', 'percent_inflam_infiltration')): 0.8471313714981079,\n",
       " (('source', 'DNP.Count.wes'),\n",
       "  ('target', 'hiv_viral_load')): 0.846625030040741,\n",
       " (('source', 'Tumor.Stage'), ('target', 'figo_stage')): 0.8456577062606812,\n",
       " (('source', 'Histologic_Type'),\n",
       "  ('target', 'hysterectomy_type')): 0.8453896045684814,\n",
       " (('source', 'Stemness.Score'),\n",
       "  ('target', 'year_of_birth')): 0.8450624942779541,\n",
       " (('source', 'integrated_subtype'),\n",
       "  ('target', 'disease_type')): 0.84502774477005,\n",
       " (('source', 'Race'), ('target', 'race')): 0.844617486000061,\n",
       " (('source', 'tumor_size_cm'),\n",
       "  ('target', 'shortest_dimension')): 0.8445382714271545,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'days_to_comorbidity')): 0.8442825078964233,\n",
       " (('source', 'ESTIMATE_stromal_score'),\n",
       "  ('target', 'age_at_onset')): 0.8440297245979309,\n",
       " (('source', 'Mutation.Count.ExcludingINDELs.wes'),\n",
       "  ('target', 'percent_monocyte_infiltration')): 0.8437231779098511,\n",
       " (('source', 'CDKN2A.mutation'), ('target', 'data_type')): 0.8431711196899414,\n",
       " (('source', 'follow_up_days'),\n",
       "  ('target', 'copy_number_estimates')): 0.8429635763168335,\n",
       " (('source', 'Ethnicity_mod'), ('target', 'aneuploidy')): 0.8424597978591919,\n",
       " (('source', 'integrated_subtype'),\n",
       "  ('target', 'specimen_type')): 0.8422080278396606,\n",
       " (('source', 'mutation_count'), ('target', 'amount')): 0.8420897722244263,\n",
       " (('source', 'Pack.Years.Smoked'),\n",
       "  ('target', 'pack_years_smoked')): 0.8418750166893005,\n",
       " (('source', 'pathologic_staging_regional_lymph_nodes_pn'),\n",
       "  ('target', 'instrument_model')): 0.8417012691497803,\n",
       " (('source', 'tumor_pathology_review'),\n",
       "  ('target', 'embolic_agent')): 0.8416364192962646,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'satellite_nodule_present')): 0.8416051268577576,\n",
       " (('source', 'follow_up_days'),\n",
       "  ('target', 'pregnancy_count')): 0.8411588668823242,\n",
       " (('source', 'BMI'),\n",
       "  ('target', 'recist_targeted_regions_sum')): 0.8410022258758545,\n",
       " (('source', 'tumor_site_curation_notes'),\n",
       "  ('target', 'programs')): 0.8406031131744385,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'per_base_sequence_quality')): 0.8402786254882812,\n",
       " (('source', 'NMF.subtype'),\n",
       "  ('target', 'simple_somatic_mutations')): 0.840055525302887,\n",
       " (('source', 'patho_staging_pt'),\n",
       "  ('target', 'ajcc_clinical_t')): 0.8399746417999268,\n",
       " (('source', 'Hypermutation_status'),\n",
       "  ('target', 'structural_variant_calling_workflows')): 0.8392250537872314,\n",
       " (('source', 'histologic_grade'),\n",
       "  ('target', 'inpc_grade')): 0.8389307260513306,\n",
       " (('source', 'TP63.cna.wes'),\n",
       "  ('target', 'schema_version')): 0.8388568162918091,\n",
       " (('source', 'weight'),\n",
       "  ('target', 'percent_normal_cells')): 0.8380669355392456,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'sarcomatoid_percent')): 0.8380641341209412,\n",
       " (('source', 'participant_country'),\n",
       "  ('target', 'country_of_birth')): 0.8371790647506714,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'undescended_testis_history')): 0.8368502855300903,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'percent_rhabdoid_features')): 0.8367599248886108,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'fragment_mean_length')): 0.8366689682006836,\n",
       " (('source', 'height'), ('target', 'shortest_dimension')): 0.8364027738571167,\n",
       " (('source', 'Tumor_Site'), ('target', 'primary_site')): 0.8363099098205566,\n",
       " (('source', 'Tumor.Stage'),\n",
       "  ('target', 'uicc_pathologic_stage')): 0.8362021446228027,\n",
       " (('source', 'participant_country'),\n",
       "  ('target', 'gleason_grade_group')): 0.8360738754272461,\n",
       " (('source', 'clinical_staging_distant_metastasis_cm'),\n",
       "  ('target', 'data_formats')): 0.835870623588562,\n",
       " (('source', 'tumor_site_curation_notes'),\n",
       "  ('target', 'cancer_detection_method')): 0.8354151248931885,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'days_to_diagnosis')): 0.8353118896484375,\n",
       " (('source', 'miRNA-seq'),\n",
       "  ('target', 'fev1_ref_post_bronch_percent')): 0.8342419862747192,\n",
       " (('source', 'integrated_subtype'),\n",
       "  ('target', 'vascular_invasion_type')): 0.8336870670318604,\n",
       " (('source', 'Sample.IDs'),\n",
       "  ('target', 'dbgap_accession_number')): 0.8334450721740723,\n",
       " (('source', 'MSI_PCR_Result'), ('target', 'slides')): 0.8329529762268066,\n",
       " (('source', 'residual_tumor'),\n",
       "  ('target', 'pediatric_kidney_staging')): 0.8321293592453003,\n",
       " (('source', 'Experiment'),\n",
       "  ('target', 'exposure_duration_hrs_per_day')): 0.8320998549461365,\n",
       " (('source', 'chr_instability_idx'),\n",
       "  ('target', 'creation_datetime')): 0.8319947719573975,\n",
       " (('source', 'BRCA2.mutation'),\n",
       "  ('target', 'expression_analysis_workflows')): 0.8319160342216492,\n",
       " (('source', 'Path_Stage_Dist_Mets-pM'),\n",
       "  ('target', 'classification_of_tumor')): 0.831291913986206,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'non_nodal_regional_disease')): 0.831098198890686,\n",
       " (('source', 'Mutation.Count.Excluding.Silent.wes'),\n",
       "  ('target', 'percent_stromal_cells')): 0.831035315990448,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'relative_smoker')): 0.8300619125366211,\n",
       " (('source', 'follow_up_days'),\n",
       "  ('target', 'relatives_with_cancer_history_count')): 0.8299530148506165,\n",
       " (('source', 'Secondhand.Smoke'),\n",
       "  ('target', 'metastasis_at_diagnosis')): 0.829879879951477,\n",
       " (('source', 'path_diag_to_death_days'),\n",
       "  ('target', 'percent_monocyte_infiltration')): 0.8288484811782837,\n",
       " (('source', 'NF1.mutation'), ('target', 'derived_files')): 0.8287805318832397,\n",
       " (('source', 'number_of_lymph_nodes_examined'),\n",
       "  ('target', 'exposure_duration_years')): 0.8279365301132202,\n",
       " (('source', 'Tumor_Size_cm'), ('target', 'msi_score')): 0.8276537656784058,\n",
       " (('source', 'CIBERSORT.Absolute.Score'),\n",
       "  ('target', 'submitted_tangent_copy_numbers')): 0.8274064064025879,\n",
       " (('source', 'NMF.subtype.core'),\n",
       "  ('target', 'simple_somatic_mutations')): 0.8270299434661865,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'per_sequence_quality_score')): 0.8269533514976501,\n",
       " (('source', 'Pack.Years.Smoked'),\n",
       "  ('target', 'percent_tumor_cells')): 0.8263245820999146,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'number_expect_cells')): 0.8253702521324158,\n",
       " (('source', 'NMF.Cluster'), ('target', 'analytes')): 0.8252964019775391,\n",
       " (('source', 'Total.Mutation.Count.wes'),\n",
       "  ('target', 'rin')): 0.8250855207443237,\n",
       " (('source', 'Tumor_Site_Other'),\n",
       "  ('target', 'tissue_or_organ_of_origin')): 0.824301540851593,\n",
       " (('source', 'ethnicity_self_identify'),\n",
       "  ('target', 'zone_of_origin_prostate')): 0.8241076469421387,\n",
       " (('source', 'KMT2D.mutation'),\n",
       "  ('target', 'data_category')): 0.8238592743873596,\n",
       " (('source', 'Subsite'),\n",
       "  ('target', 'method_of_sample_procurement')): 0.8231463432312012,\n",
       " (('source', 'MAP3K1.Mutation.Type'), ('target', 'files')): 0.8224369287490845,\n",
       " (('source', 'Ethnicity'), ('target', 'race')): 0.8218976259231567,\n",
       " (('source', 'num_yrs_alc_con'),\n",
       "  ('target', 'sarcomatoid_percent')): 0.8218024969100952,\n",
       " (('source', 'Case_ID'), ('target', 'mirna_expressions')): 0.8214774131774902,\n",
       " (('source', 'ESTIMATE.TumorPurity.rna'),\n",
       "  ('target', 'file_size')): 0.8213016986846924,\n",
       " (('source', 'ImmuneCluster.rna'), ('target', 'samples')): 0.8210800886154175,\n",
       " (('source', 'TMT.Plex'), ('target', 'number_of_cycles')): 0.820878267288208,\n",
       " (('source', 'MLLT4.Mutation.Type'),\n",
       "  ('target', 'adapter_name')): 0.82044517993927,\n",
       " (('source', 'smoking_second_hand'),\n",
       "  ('target', 'metastasis_at_diagnosis')): 0.8200252056121826,\n",
       " (('source', 'histologic_grade'),\n",
       "  ('target', 'weiss_assessment_findings')): 0.8196114301681519,\n",
       " (('source', 'P16'), ('target', 'notes')): 0.819571852684021,\n",
       " (('source', 'Path_Stage_Primary_Tumor-pT'),\n",
       "  ('target', 'adapter_sequence')): 0.8188292384147644,\n",
       " (('source', 'GATA3.Mutation.Type'), ('target', 'state')): 0.8184850215911865,\n",
       " (('source', 'NMF.Cluster'), ('target', 'intron')): 0.818397581577301,\n",
       " (('source', 'tumor_site_curated'),\n",
       "  ('target', 'max_tumor_bulk_site')): 0.8180767297744751,\n",
       " (('source', 'weight'),\n",
       "  ('target', 'fev1_fvc_pre_bronch_percent')): 0.8174716830253601,\n",
       " (('source', 'tumor_site_curation_notes'),\n",
       "  ('target', 'occupation_type')): 0.8168277740478516,\n",
       " (('source', 'histologic_type'),\n",
       "  ('target', 'described_cases')): 0.815576434135437,\n",
       " (('source', 'Ethnicity_Self_Identify'),\n",
       "  ('target', 'esophageal_columnar_dysplasia_degree')): 0.8145716190338135,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'contamination_error')): 0.8145550489425659,\n",
       " (('source', 'Clin_Stage_Dist_Mets-cM'),\n",
       "  ('target', 'clonality')): 0.814075767993927,\n",
       " (('source', 'Path_Stage_Primary_Tumor_pT'),\n",
       "  ('target', 'histone_family')): 0.8140029907226562,\n",
       " (('source', 'TP53.mutation'),\n",
       "  ('target', 'derived_files')): 0.8139206171035767,\n",
       " (('source', 'ARID1A.Mutation.Type'),\n",
       "  ('target', 'projects')): 0.8130854368209839,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'treatment_dose')): 0.8123106956481934,\n",
       " (('source', 'BRCA2.mutation'),\n",
       "  ('target', 'specialized_molecular_test')): 0.8122996091842651,\n",
       " (('source', 'CMS'), ('target', 'base_caller_name')): 0.8120805025100708,\n",
       " (('source', 'PT'), ('target', 'ajcc_pathologic_t')): 0.8117151260375977,\n",
       " (('source', 'Country'), ('target', 'country_of_birth')): 0.8109898567199707,\n",
       " (('source', 'Neoplastic_cellularity'),\n",
       "  ('target', 'chromosomal_translocation')): 0.810788631439209,\n",
       " (('source', 'Tumor_Size_cm'), ('target', 'tumor_burden')): 0.8105449676513672,\n",
       " (('source', 'xCell.Stromal.Score'),\n",
       "  ('target', 'days_to_treatment_start')): 0.8104717135429382,\n",
       " (('source', 'MSI_PCR_Result'),\n",
       "  ('target', 'somatic_copy_number_workflows')): 0.8102229833602905,\n",
       " (('source', 'medical_condition'),\n",
       "  ('target', 'relationship_primary_diagnosis')): 0.8098931312561035,\n",
       " (('source', 'tumor_site'),\n",
       "  ('target', 'imaging_anatomic_site')): 0.8098634481430054,\n",
       " (('source', 'Hypermutation_status'),\n",
       "  ('target', 'alignment_workflows')): 0.8096431493759155,\n",
       " (('source', 'pathologic_staging_regional_lymph_nodes_pn'),\n",
       "  ('target', 'platforms')): 0.8092175126075745,\n",
       " (('source', 'Histologic_type'),\n",
       "  ('target', 'margins_involved_site')): 0.8087670803070068,\n",
       " (('source', 'Experiment'), ('target', 'tumor_burden')): 0.8078441619873047,\n",
       " (('source', 'case_id'),\n",
       "  ('target', 'pathology_report_uuid')): 0.8078262805938721,\n",
       " (('source', 'TP63.cna.wes'),\n",
       "  ('target', 'base_caller_version')): 0.8070162534713745,\n",
       " (('source', 'Participant'),\n",
       "  ('target', 'pathology_report_uuid')): 0.8067994117736816,\n",
       " (('source', 'CUL3.mutation'),\n",
       "  ('target', 'experimental_strategy')): 0.8066204786300659,\n",
       " (('source', 'ERBB2.Gene.Amplified'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'TOP2A.Gene.Amplified'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'TP53.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'PTEN.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'AKT1.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'CBFB.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'SF3B1.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'ARID1A.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'MLLT4.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'PTEN.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'KMT2D.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'CUL3.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'BRCA2.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'KEAP1.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'NF1.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'NOTCH1.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'CUL3.NFE2L2.KEAP1.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'CDKN2A.pathway.alteration.status'),\n",
       "  ('target', 'wgs_coverage')): 0.8063391447067261,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'rhabdoid_present')): 0.8052890300750732,\n",
       " (('source', 'pathologic_staging_regional_lymph_nodes_pn'),\n",
       "  ('target', 'encoding')): 0.804939866065979,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'average_insert_size')): 0.804742693901062,\n",
       " (('source', 'Number.of.non.synonymous.Mutations'),\n",
       "  ('target', 'days_to_collection')): 0.8047115206718445,\n",
       " (('source', 'Neoplastic_cellularity'),\n",
       "  ('target', 'weeks_gestation_at_birth')): 0.8045267462730408,\n",
       " (('source', 'Cigarettes.per.Day'),\n",
       "  ('target', 'years_smoked')): 0.8040656447410583,\n",
       " (('source', 'ARID1A.mutation'),\n",
       "  ('target', 'spike_ins_fasta')): 0.8040547966957092,\n",
       " (('source', 'tumor_stage_pathological'),\n",
       "  ('target', 'uicc_clinical_stage')): 0.803923487663269,\n",
       " (('source', 'pathologic_staging_distant_metastasis_pm'),\n",
       "  ('target', 'ensat_pathologic_n')): 0.80373215675354,\n",
       " (('source', 'NOTCH1.mutation'),\n",
       "  ('target', 'experimental_strategy')): 0.8036531209945679,\n",
       " (('source', 'Smoking.Signature.Count.wes'),\n",
       "  ('target', 'fragment_maximum_length')): 0.8032849431037903,\n",
       " (('source', 'ImmuneCluster.rna'),\n",
       "  ('target', 'parent_samples')): 0.8025866150856018,\n",
       " (('source', 'Total.Mutation.Count.per.Mb.wes'),\n",
       "  ('target', 'days_to_last_known_disease_status')): 0.8024271726608276,\n",
       " (('source', 'KMT2D.mutation'),\n",
       "  ('target', 'genomic_profile_harmonization_workflows')): 0.8024219274520874,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'overrepresented_sequences')): 0.8022950887680054,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'days_to_sequencing')): 0.8022382259368896,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'days_to_sequencing')): 0.8019066452980042,\n",
       " (('source', 'clinic_staging_dist_metas'),\n",
       "  ('target', 'clonality')): 0.8018298149108887,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'passage_count')): 0.8016184568405151,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'passage_count')): 0.8011714816093445,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'passage_count')): 0.8010097742080688,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'hormonal_contraceptive_type')): 0.8009200096130371,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'hormonal_contraceptive_type')): 0.800545334815979,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'passage_count')): 0.8004685044288635,\n",
       " (('source', 'CIBERSORT.Absolute.Score'),\n",
       "  ('target', 'batch_id')): 0.8003429174423218,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'rhabdoid_percent')): 0.8002649545669556,\n",
       " (('source', 'num_yrs_alc_con'),\n",
       "  ('target', 'exposure_duration_years')): 0.800102949142456,\n",
       " (('source', 'path_diag_to_death_days'),\n",
       "  ('target', 'percent_eosinophil_infiltration')): 0.8000715970993042,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'days_to_sequencing')): 0.8000017404556274,\n",
       " (('source', 'Clin_Stage_Dist_Mets-cM'),\n",
       "  ('target', 'burkitt_lymphoma_clinical_variant')): 0.7999782562255859,\n",
       " (('source', 'CD3_IHC_count'),\n",
       "  ('target', 'percent_tumor_cells')): 0.7996212840080261,\n",
       " (('source', 'clinic_staging_dist_metas'),\n",
       "  ('target', 'viral_hepatitis_serologies')): 0.7994539737701416,\n",
       " (('source', 'num_yrs_alc_con'),\n",
       "  ('target', 'percent_rhabdoid_features')): 0.799392580986023,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'percent_sarcomatoid_features')): 0.7993564009666443,\n",
       " (('source', 'TMT.Channel'),\n",
       "  ('target', 'fab_morphology_code')): 0.7989867925643921,\n",
       " (('source', 'PIK3CA.Mutation.Type'),\n",
       "  ('target', 'copy_number_auxiliary_files')): 0.7987931966781616,\n",
       " (('source', 'Total.Mutation.Count.per.Mb.wes'),\n",
       "  ('target', 'tumor_depth_measurement')): 0.7986514568328857,\n",
       " (('source', 'num_yrs_alc_con'),\n",
       "  ('target', 'rhabdoid_percent')): 0.7985134124755859,\n",
       " (('source', 'TP53.Mutation.Type'),\n",
       "  ('target', 'genomic_profile_harmonization_workflows')): 0.7984115481376648,\n",
       " (('source', 'PIK3CA.Mutation.Type'),\n",
       "  ('target', 'reason_treatment_not_given')): 0.7983793020248413,\n",
       " (('source', 'TP53.mutation.status'),\n",
       "  ('target', 'days_to_recurrence')): 0.798122763633728,\n",
       " (('source', 'tumor_reoccur_after_treatment'),\n",
       "  ('target', 'days_to_recurrence')): 0.798122763633728,\n",
       " (('source', 'Smoking.Signature.Fraction.wes'),\n",
       "  ('target', 'year_of_diagnosis')): 0.7981029748916626,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'days_to_sequencing')): 0.7975791692733765,\n",
       " (('source', 'PTEN.mutation'), ('target', 'adapter_name')): 0.7968547344207764,\n",
       " (('source', 'Pack.Years.Smoked'),\n",
       "  ('target', 'initial_weight')): 0.7967960834503174,\n",
       " (('source', 'Tumor_Size_cm'),\n",
       "  ('target', 'greatest_tumor_dimension')): 0.7966986894607544,\n",
       " (('source', 'PIK3CA.pathway.alteration'),\n",
       "  ('target', 'gene_expressions')): 0.7957477569580078,\n",
       " (('source', 'Label-free Proteome'),\n",
       "  ('target', 'tumor_infiltrating_macrophages')): 0.7951736450195312,\n",
       " (('source', 'NF1.mutation'),\n",
       "  ('target', 'run_metadata_files')): 0.7945616245269775,\n",
       " (('source', 'bmi'), ('target', 'bmi')): 0.7945448160171509,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes_pN'),\n",
       "  ('target', 'encoding')): 0.7945354580879211,\n",
       " (('source', 'Participant'),\n",
       "  ('target', 'catalog_reference')): 0.7942465543746948,\n",
       " (('source', 'Tumor.Status'), ('target', 'index_date')): 0.7941085696220398,\n",
       " (('source', 'MSH2_mutation'),\n",
       "  ('target', 'tumor_infiltrating_macrophages')): 0.7940020561218262,\n",
       " (('source', 'MSH6_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.7939920425415039,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'days_to_adverse_event')): 0.792889416217804,\n",
       " (('source', 'Experiment'), ('target', 'test_value')): 0.7928625345230103,\n",
       " (('source', 'TMT Proteome (Normal)'),\n",
       "  ('target', 'treatment_dose_units')): 0.7922723293304443,\n",
       " (('source', 'TP53.mutation.status'),\n",
       "  ('target', 'icd_10_code')): 0.7919824719429016,\n",
       " (('source', 'tumor_reoccur_after_treatment'),\n",
       "  ('target', 'icd_10_code')): 0.7919824719429016,\n",
       " (('source', 'PTEN.Mutation.Type'),\n",
       "  ('target', 'library_preparation_kit_version')): 0.7914172410964966,\n",
       " (('source', 'Histologic_type'),\n",
       "  ('target', 'morphologic_architectural_pattern')): 0.7907872200012207,\n",
       " (('source', 'tumor_pathology_review'),\n",
       "  ('target', 'scan_tracer_used')): 0.7905125617980957,\n",
       " (('source', 'TMT Phosphoproteome (Normal)'),\n",
       "  ('target', 'treatment_dose_units')): 0.7904951572418213,\n",
       " (('source', 'case_id'), ('target', 'catalog_reference')): 0.7903698682785034,\n",
       " (('source', 'tumor_included_for_the_study'),\n",
       "  ('target', 'relative_with_cancer_history')): 0.7902082800865173,\n",
       " (('source', 'NOTCH1.mutation'),\n",
       "  ('target', 'genomic_profile_harmonization_workflows')): 0.7899852991104126,\n",
       " (('source', 'TP53.Mutation.Type'),\n",
       "  ('target', 'experimental_strategy')): 0.7893421649932861,\n",
       " (('source', 'Clin_Stage_Dist_Mets-cM'),\n",
       "  ('target', 'histologic_progression_type')): 0.7892955541610718,\n",
       " (('source', 'integrated_subtype'),\n",
       "  ('target', 'library_selection')): 0.7891408205032349,\n",
       " (('source', 'TMT Phosphoproteome'),\n",
       "  ('target', 'mitotic_count')): 0.7891157865524292,\n",
       " (('source', 'bmi'), ('target', 'percent_stromal_cells')): 0.7890137434005737,\n",
       " (('source', 'miRNA-seq'),\n",
       "  ('target', 'fev1_ref_pre_bronch_percent')): 0.7890126705169678,\n",
       " (('source', 'Label-free Proteome'),\n",
       "  ('target', 'tumor_infiltrating_lymphocytes')): 0.7883134484291077,\n",
       " (('source', 'Mutation.Count.ExcludingINDELs.wes'),\n",
       "  ('target', 'greatest_tumor_dimension')): 0.7880262136459351,\n",
       " (('source', 'patho_staging_pt'),\n",
       "  ('target', 'ajcc_pathologic_t')): 0.7878679037094116,\n",
       " (('source', 'number_of_lymph_nodes_positive_for_tumor'),\n",
       "  ('target', 'tumor_thickness')): 0.7877035140991211,\n",
       " (('source', 'MSH2_mutation'),\n",
       "  ('target', 'tumor_infiltrating_lymphocytes')): 0.7872805595397949,\n",
       " (('source', 'Fat_fraction'),\n",
       "  ('target', 'peripancreatic_lymph_nodes_positive')): 0.7867388129234314,\n",
       " (('source', 'BMI'), ('target', 'longest_dimension')): 0.7860113978385925,\n",
       " (('source', 'ESTIMATE.TumorPurity'),\n",
       "  ('target', 'average_read_length')): 0.7858943939208984,\n",
       " (('source', 'KAT6A.cna.wes'),\n",
       "  ('target', 'schema_version')): 0.7850717306137085,\n",
       " (('source', 'Country.of.Origin'),\n",
       "  ('target', 'chemical_exposure_type')): 0.7849422097206116,\n",
       " (('source', 'PIK3CA.pathway.alteration.status'),\n",
       "  ('target', 'icd_10_code')): 0.7846299409866333,\n",
       " (('source', 'NRAS_mutation'),\n",
       "  ('target', 'tumor_regression_grade')): 0.7838961482048035,\n",
       " (('source', 'number_of_lymph_nodes_positive_for_tumor'),\n",
       "  ('target', 'margin_distance')): 0.7833651304244995,\n",
       " (('source', 'FGFR1.cna.wes'),\n",
       "  ('target', 'schema_version')): 0.7833024859428406,\n",
       " (('source', 'gender'),\n",
       "  ('target', 'ovarian_specimen_status')): 0.7828259468078613,\n",
       " (('source', 'expert_path_review'),\n",
       "  ('target', 'per_tile_sequence_quality')): 0.7823749780654907,\n",
       " (('source', 'path_diag_to_tumor_reoccur_days'),\n",
       "  ('target', 'treatment_duration')): 0.7820603251457214,\n",
       " (('source', 'Gender'),\n",
       "  ('target', 'ovarian_specimen_status')): 0.7819461822509766,\n",
       " (('source', 'Number.of.non.synonymous.Mutations'),\n",
       "  ('target', 'days_to_imaging')): 0.7818870544433594,\n",
       " (('source', 'CUL3.mutation'),\n",
       "  ('target', 'proc_internal')): 0.7817186117172241,\n",
       " (('source', 'sex'),\n",
       "  ('target', 'ovarian_specimen_status')): 0.7814369201660156,\n",
       " (('source', 'TMT Proteome'), ('target', 'mitotic_count')): 0.7811578512191772,\n",
       " (('source', 'xCell.MicroenvironmentScore.rna'),\n",
       "  ('target', 'proportion_reads_duplicated')): 0.7809293270111084,\n",
       " (('source', 'CIN.wes'), ('target', 'number_expect_cells')): 0.78024822473526,\n",
       " (('source', 'ESTIMATE.TumorPurity.rna'),\n",
       "  ('target', 'workflow_link')): 0.7795905470848083,\n",
       " (('source', 'PTEN.mutation'),\n",
       "  ('target', 'submitted_genomic_profiles')): 0.7794400453567505,\n",
       " (('source', 'Path_Stage_Dist_Mets-pM'),\n",
       "  ('target', 'tumor_descriptor')): 0.7790616154670715,\n",
       " (('source', 'tumor_focality'),\n",
       "  ('target', 'progression_or_recurrence_type')): 0.7781952619552612,\n",
       " (('source', 'country'), ('target', 'gleason_grade_group')): 0.778130829334259,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'diabetes_treatment_type')): 0.7780377268791199,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'diabetes_treatment_type')): 0.777965247631073,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'proportion_reads_duplicated')): 0.7777878046035767,\n",
       " (('source', 'Tumor_Focality'),\n",
       "  ('target', 'progression_or_recurrence_type')): 0.7776193618774414,\n",
       " (('source', 'CIMP.status.meth'),\n",
       "  ('target', 'medulloblastoma_molecular_classification')): 0.7774020433425903,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_base_sequence_content')): 0.7772347331047058,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'hormone_replacement_therapy_type')): 0.7769582271575928,\n",
       " (('source', 'GATA3.Mutation.Status'),\n",
       "  ('target', 'platform')): 0.7768841981887817,\n",
       " (('source', 'ER.Updated.Clinical.Status'),\n",
       "  ('target', 'eln_risk_classification')): 0.7765294313430786,\n",
       " (('source', 'TCGA.RNA.cluster'),\n",
       "  ('target', 'extracapsular_extension')): 0.7758499383926392,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'hormone_replacement_therapy_type')): 0.7758058309555054,\n",
       " (('source', 'TOP2A.Proteogenomic.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.7753781080245972,\n",
       " (('source', 'NOTCH1.mutation'),\n",
       "  ('target', 'adapter_name')): 0.7753483057022095,\n",
       " (('source', 'height'), ('target', 'analyte_quantity')): 0.7752541303634644,\n",
       " (('source', 'CDKN2A.mutation.status'),\n",
       "  ('target', 'platform')): 0.7752410173416138,\n",
       " (('source', 'NFE2L2.mutation'),\n",
       "  ('target', 'read_group_name')): 0.7748939394950867,\n",
       " (('source', 'PIK3CA.mutation'),\n",
       "  ('target', 'read_group_name')): 0.7748939394950867,\n",
       " (('source', 'AKT1.Mutation.Type'),\n",
       "  ('target', 'read_group_name')): 0.7748938798904419,\n",
       " (('source', 'KEAP1.mutation'),\n",
       "  ('target', 'read_group_name')): 0.7748938798904419,\n",
       " (('source', 'MSI_PCR_Result'),\n",
       "  ('target', 'workflow_type')): 0.7746422290802002,\n",
       " (('source', 'TMT Phosphoproteome'),\n",
       "  ('target', 'lesions_treated_number')): 0.7740669250488281,\n",
       " (('source', 'Stemness.Score'),\n",
       "  ('target', 'tobacco_smoking_onset_year')): 0.7736896872520447,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'asbestos_exposure')): 0.773684024810791,\n",
       " (('source', 'TP53.mutation.status'),\n",
       "  ('target', 'icd_10')): 0.7734252214431763,\n",
       " (('source', 'tumor_reoccur_after_treatment'),\n",
       "  ('target', 'icd_10')): 0.7734252214431763,\n",
       " (('source', 'SUZ12.mutation.status'),\n",
       "  ('target', 'mismatch_repair_mutation')): 0.7733807563781738,\n",
       " (('source', 'FGFR3.TACC3.fusion.rna'),\n",
       "  ('target', 'mismatch_repair_mutation')): 0.773380696773529,\n",
       " (('source', 'PIK3CA.Mutation.Status'),\n",
       "  ('target', 'platform')): 0.7733219861984253,\n",
       " (('source', 'xCell.MicroenvironmentScore.rna'),\n",
       "  ('target', 'proportion_targets_no_coverage')): 0.7729025483131409,\n",
       " (('source', 'Stromal_fraction'),\n",
       "  ('target', 'tumor_code_id')): 0.7728467583656311,\n",
       " (('source', 'vital_status'),\n",
       "  ('target', 'marital_status')): 0.7724044322967529,\n",
       " (('source', 'ERBB2.Updated.Clinical.Status'),\n",
       "  ('target', 'eln_risk_classification')): 0.7720450162887573,\n",
       " (('source', 'PIK3CA.pathway.alteration.status'),\n",
       "  ('target', 'days_to_recurrence')): 0.7719029188156128,\n",
       " (('source', 'MSH2_mutation'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7716974020004272,\n",
       " (('source', 'CD3_IHC_count'), ('target', 'concentration')): 0.77155601978302,\n",
       " (('source', 'CIBERSORT.Absolute.Score'),\n",
       "  ('target', 'days_to_last_follow_up')): 0.7715497612953186,\n",
       " (('source', 'stemness_score'),\n",
       "  ('target', 'year_of_death')): 0.7712750434875488,\n",
       " (('source', 'DNP.Count.GG.to.TT.or.CC.to.AA.wes'),\n",
       "  ('target', 'legacy_created_datetime')): 0.7707582712173462,\n",
       " (('source', 'clinic_staging_dist_metas'),\n",
       "  ('target', 'enneking_msts_metastasis')): 0.7705752849578857,\n",
       " (('source', 'expert_path_review_details'),\n",
       "  ('target', 'therapeutic_agents')): 0.7703258991241455,\n",
       " (('source', 'Total.Mutation.Count.wes'),\n",
       "  ('target', 'tumor_width_measurement')): 0.7702203989028931,\n",
       " (('source', 'Subsite'),\n",
       "  ('target', 'site_of_resection_or_biopsy')): 0.7700804471969604,\n",
       " (('source', 'CDKN2A.cna.wes'),\n",
       "  ('target', 'histone_variant')): 0.7694271802902222,\n",
       " (('source', 'TNBC.Updated.Clinical.Status'),\n",
       "  ('target', 'eln_risk_classification')): 0.7690706253051758,\n",
       " (('source', 'PIK3CA.pathway.alteration.status'),\n",
       "  ('target', 'platform')): 0.7690633535385132,\n",
       " (('source', 'ARID1A.mutation'), ('target', 'plate_name')): 0.7687243223190308,\n",
       " (('source', 'patho_staging_pn'),\n",
       "  ('target', 'gene_symbol')): 0.7686645984649658,\n",
       " (('source', 'ImmuneScore'),\n",
       "  ('target', 'size_selection_range')): 0.7686562538146973,\n",
       " (('source', 'Experiment'), ('target', 'imaging_suv_max')): 0.7686212062835693,\n",
       " (('source', 'Label-free Proteome'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7684435844421387,\n",
       " (('source', 'TMT Phosphoproteome'),\n",
       "  ('target', 'loci_abnormal_count')): 0.7684091925621033,\n",
       " (('source', 'tumor_site'),\n",
       "  ('target', 'section_location')): 0.7681102156639099,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'fragment_maximum_length')): 0.7678172588348389,\n",
       " (('source', 'country_of_origin'),\n",
       "  ('target', 'gleason_grade_group')): 0.7678132057189941,\n",
       " (('source', 'ESTIMATE.Stromal.Score'),\n",
       "  ('target', 'days_to_last_known_disease_status')): 0.7671908140182495,\n",
       " (('source', 'TMT Proteome'),\n",
       "  ('target', 'loci_abnormal_count')): 0.7667060494422913,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'consistent_pathology_review')): 0.7665508985519409,\n",
       " (('source', 'TMT Proteome'),\n",
       "  ('target', 'lesions_treated_number')): 0.7663989067077637,\n",
       " (('source', 'TMT Phosphoproteome'),\n",
       "  ('target', 'lymph_nodes_tested')): 0.7657071352005005,\n",
       " (('source', 'ESTIMATE.Stromal.Score'),\n",
       "  ('target', 'days_to_birth')): 0.7656859755516052,\n",
       " (('source', 'Neoplastic_cellularity'),\n",
       "  ('target', 'spindle_cell_percent')): 0.765556812286377,\n",
       " (('source', 'POLE_mutation'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7652941942214966,\n",
       " (('source', 'MLLT4.Mutation.Type'),\n",
       "  ('target', 'read_groups')): 0.764785885810852,\n",
       " (('source', 'Neoplastic_cellularity'),\n",
       "  ('target', 'hpv_positive_type')): 0.7644059658050537,\n",
       " (('source', 'ESTIMATE.ImmuneScore.rna'),\n",
       "  ('target', 'average_insert_size')): 0.7641798257827759,\n",
       " (('source', 'PIK3CA.pathway.alteration'),\n",
       "  ('target', 'submitter_id')): 0.7641223669052124,\n",
       " (('source', 'pathologic_staging_distant_metastasis_pm'),\n",
       "  ('target', 'residual_disease')): 0.7639073133468628,\n",
       " (('source', 'TMT Proteome'),\n",
       "  ('target', 'lymph_nodes_tested')): 0.7635756134986877,\n",
       " (('source', 'PIK3CA.pathway.alteration.status'),\n",
       "  ('target', 'icd_10')): 0.763058066368103,\n",
       " (('source', 'CD3.TILS.Status'),\n",
       "  ('target', 'per_base_n_content')): 0.7630445957183838,\n",
       " (('source', 'KRAS_mutation'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7628421783447266,\n",
       " (('source', 'NMF.Cluster'),\n",
       "  ('target', 'protein_expressions')): 0.7627841234207153,\n",
       " (('source', 'ESTIMATE.StromalScore.rna'),\n",
       "  ('target', 'mitotic_total_area')): 0.7623012661933899,\n",
       " (('source', 'Total.Mutation.Count.wes'),\n",
       "  ('target', 'greatest_tumor_dimension')): 0.7620115876197815,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'double_hit_lymphoma')): 0.7619484066963196,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'parent_with_radiation_exposure')): 0.7619277238845825,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'nadir_cd4_count')): 0.7618865966796875,\n",
       " (('source', 'TP53.Mutation.Type'),\n",
       "  ('target', 'plate_name')): 0.761695384979248,\n",
       " (('source', 'transcriptomic_subtype'),\n",
       "  ('target', 'ploidy')): 0.7614459991455078,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'exposure_duration_years')): 0.760873019695282,\n",
       " (('source', 'histology_diagnosis'),\n",
       "  ('target', 'risk_factors')): 0.7608436346054077,\n",
       " (('source', 'ERBB2.Proteogenomic.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.7608059644699097,\n",
       " (('source', 'number_of_lymph_nodes_examined'),\n",
       "  ('target', 'occupation_duration_years')): 0.7604855895042419,\n",
       " (('source', 'Clin_Stage_Dist_Mets-cM'),\n",
       "  ('target', 'largest_extrapelvic_peritoneal_focus')): 0.7600830793380737,\n",
       " (('source', 'Total.Mutation.Count.per.Mb.wes'),\n",
       "  ('target', 'msi_score')): 0.7596982717514038,\n",
       " (('source', 'Chromosome.INstability.index.CIN.'),\n",
       "  ('target', 'treatment_dose')): 0.7594877481460571,\n",
       " (('source', 'tumor_site'),\n",
       "  ('target', 'tissue_source_sites')): 0.7591565251350403,\n",
       " (('source', 'PIK3CA.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.7590274810791016,\n",
       " (('source', 'Experiment'),\n",
       "  ('target', 'tumor_largest_dimension_diameter')): 0.7586588263511658,\n",
       " (('source', 'Sample.IDs'),\n",
       "  ('target', 'multiplex_barcode')): 0.7586580514907837,\n",
       " (('source', 'Number.of.non.synonymous.Mutations'),\n",
       "  ('target', 'days_to_adverse_event')): 0.7582264542579651,\n",
       " (('source', 'xCell.Immune.Score'),\n",
       "  ('target', 'days_to_treatment_end')): 0.7579348683357239,\n",
       " (('source', 'num_yrs_alc_con'),\n",
       "  ('target', 'percent_sarcomatoid_features')): 0.7575830221176147,\n",
       " (('source', 'TP53.Mutation.Type'),\n",
       "  ('target', 'adapter_name')): 0.7571585178375244,\n",
       " (('source', 'NMF.Cluster.Membership.Score'),\n",
       "  ('target', 'biospecimen_volume')): 0.7560111284255981,\n",
       " (('source', 'CDKN2A.mutation.status'),\n",
       "  ('target', 'wgs_coverage')): 0.7559474110603333,\n",
       " (('source', 'pathologic_staging_primary_tumor_pt'),\n",
       "  ('target', 'who_cns_grade')): 0.7558034658432007,\n",
       " (('source', 'POLE_mutation'), ('target', 'loci_count')): 0.7557411193847656,\n",
       " (('source', 'country'),\n",
       "  ('target', 'country_of_residence_at_enrollment')): 0.7556219100952148,\n",
       " (('source', 'ESTIMATE_stromal_score'),\n",
       "  ('target', 'percent_gc_content')): 0.7555307149887085,\n",
       " (('source', 'Fat_fraction'),\n",
       "  ('target', 'tobacco_smoking_status')): 0.755292534828186,\n",
       " (('source', 'BMI'),\n",
       "  ('target', 'fragment_standard_deviation_length')): 0.7552276849746704,\n",
       " (('source', 'alcohol_consumption'),\n",
       "  ('target', 'reflux_treatment_type')): 0.7550856471061707,\n",
       " (('source', 'Case_ID'),\n",
       "  ('target', 'pathology_report_uuid')): 0.754883885383606,\n",
       " (('source', 'NF1.mutation'), ('target', 'files')): 0.7544796466827393,\n",
       " (('source', 'PR.Clinical.Status'),\n",
       "  ('target', 'lymph_node_involvement')): 0.7543972730636597,\n",
       " (('source', 'CD3_IHC_count'),\n",
       "  ('target', 'pack_years_smoked')): 0.7541168928146362,\n",
       " (('source', 'tumor_size_cm'), ('target', 'imaging_suv')): 0.7540547847747803,\n",
       " (('source', 'alcohol_consum'),\n",
       "  ('target', 'reflux_treatment_type')): 0.7535959482192993,\n",
       " (('source', 'KRAS_mutation'), ('target', 'loci_count')): 0.7535419464111328,\n",
       " (('source', 'PIK3CA.pathway.alteration.status'),\n",
       "  ('target', 'chip_id')): 0.7531353235244751,\n",
       " (('source', 'CBFB.Mutation.Type'),\n",
       "  ('target', 'sequencing_center')): 0.753021240234375,\n",
       " (('source', 'Tumor_Size_cm'),\n",
       "  ('target', 'tumor_depth_measurement')): 0.7527634501457214,\n",
       " (('source', 'SampleID'),\n",
       "  ('target', 'protocol_identifier')): 0.7523972988128662,\n",
       " (('source', 'histologic_grade'),\n",
       "  ('target', 'tissue_type')): 0.7517302632331848,\n",
       " (('source', 'Path_Stage_Reg_Lymph_Nodes-pN'),\n",
       "  ('target', 'platforms')): 0.7513548135757446,\n",
       " (('source', 'UMS'), ('target', 'tumor_code')): 0.7510716915130615,\n",
       " (('source', 'NMF.subtype.core'),\n",
       "  ('target', 'experimental_protocol_type')): 0.7508062124252319,\n",
       " (('source', 'NMF.subtype.membership.score'),\n",
       "  ('target', 'proportion_targets_no_coverage')): 0.7504590749740601,\n",
       " (('source', 'CBFB.Mutation.Type'),\n",
       "  ('target', 'data_types')): 0.7502865791320801,\n",
       " (('source', 'secondhand_smoke_exposure'),\n",
       "  ('target', 'international_prognostic_index')): 0.7502639889717102,\n",
       " (('source', 'AKT1.Mutation.Type'),\n",
       "  ('target', 'first_event')): 0.7501581907272339,\n",
       " (('source', 'NFE2L2.mutation'),\n",
       "  ('target', 'first_event')): 0.7501581907272339,\n",
       " (('source', 'KEAP1.mutation'), ('target', 'first_event')): 0.7501581907272339,\n",
       " (('source', 'PIK3CA.mutation'),\n",
       "  ('target', 'first_event')): 0.7501581907272339,\n",
       " (('source', 'GATA3.Mutation.Status'),\n",
       "  ('target', 'wgs_coverage')): 0.7497786283493042,\n",
       " (('source', 'TCGA.RNA.cluster'),\n",
       "  ('target', 'tumor_shape')): 0.749733567237854,\n",
       " (('source', 'Tumor_Size_cm'),\n",
       "  ('target', 'imaging_suv_max')): 0.7496187686920166,\n",
       " (('source', 'Vital.Status'), ('target', 'vital_status')): 0.7494643926620483,\n",
       " (('source', 'Path_Stage_Primary_Tumor_pT'),\n",
       "  ('target', 'chipseq_antibody')): 0.7488166093826294,\n",
       " (('source', 'ESTIMATE.Immune.Score'),\n",
       "  ('target', 'number_expect_cells')): 0.7486101388931274,\n",
       " (('source', 'ESTIMATE_stromal_score'),\n",
       "  ('target', 'prostatic_involvement_percent')): 0.7484387159347534,\n",
       " (('source', 'xCell.MicroenvironmentScore.rna'),\n",
       "  ('target', 'proportion_coverage_10x')): 0.7480713129043579,\n",
       " (('source', 'KAT6A.cna.wes'),\n",
       "  ('target', 'base_caller_version')): 0.747939944267273,\n",
       " (('source', 'TP53.mutation.status'),\n",
       "  ('target', 'platform')): 0.7475174069404602,\n",
       " (('source', 'tumor_reoccur_after_treatment'),\n",
       "  ('target', 'platform')): 0.7475174069404602,\n",
       " (('source', 'ESTIMATE_immune_score'),\n",
       "  ('target', 'pairs_on_diff_chr')): 0.7470560073852539,\n",
       " (('source', 'FGFR1.cna.wes'),\n",
       "  ('target', 'base_caller_version')): 0.7466878890991211,\n",
       " (('source', 'HPV_inference'),\n",
       "  ('target', 'history_of_tumor')): 0.7465462684631348,\n",
       " (('source', 'MSH2_mutation'),\n",
       "  ('target', 'mitotic_count')): 0.7464016675949097,\n",
       " (('source', 'Total.Mutation.Count.per.Mb.wes'),\n",
       "  ('target', 'imaging_suv')): 0.7463942766189575,\n",
       " (('source', 'TMT Phosphoproteome'),\n",
       "  ('target', 'use_per_day')): 0.7461803555488586,\n",
       " (('source', 'Participant'), ('target', 'code')): 0.7461411356925964,\n",
       " (('source', 'num_pack_years_sm'),\n",
       "  ('target', 'percent_necrosis')): 0.7452598214149475,\n",
       " (('source', 'path_diag_to_tumor_reoccur_days'),\n",
       "  ('target', 'amount')): 0.7448740005493164,\n",
       " (('source', 'TMT Proteome'), ('target', 'use_per_day')): 0.7446928024291992,\n",
       " (('source', 'BRAF_mutation'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7437111139297485,\n",
       " (('source', 'Vital.Status'),\n",
       "  ('target', 'tissue_collection_type')): 0.7431253790855408,\n",
       " (('source', 'Chromosome.INstability.index.CIN.'),\n",
       "  ('target', 'magnification')): 0.7431067228317261,\n",
       " (('source', 'NMF.subtype'),\n",
       "  ('target', 'experimental_protocol_type')): 0.7428311705589294,\n",
       " (('source', 'Ischemia.Time.in.Minutes'),\n",
       "  ('target', 'margin_distance')): 0.7424702644348145,\n",
       " (('source', 'Country.of.Origin'),\n",
       "  ('target', 'asbestos_exposure_type')): 0.7423560619354248,\n",
       " (('source', 'miRNA-seq'), ('target', 'growth_rate')): 0.7423207759857178,\n",
       " (('source', 'MSI_PCR_Result'),\n",
       "  ('target', 'oct_embedded')): 0.7414677143096924,\n",
       " (('source', 'PTEN.Mutation.Type'),\n",
       "  ('target', 'library_preparation_kit_name')): 0.7412052154541016,\n",
       " (('source', 'ESTIMATEScore'),\n",
       "  ('target', 'size_selection_range')): 0.7408730983734131,\n",
       " (('source', 'QC.status'), ('target', 'basic_statistics')): 0.7407190799713135,\n",
       " (('source', 'PMS2_mutation'),\n",
       "  ('target', 'cigarettes_per_day')): 0.7406757473945618,\n",
       " (('source', 'StromalScore'),\n",
       "  ('target', 'size_selection_range')): 0.7404661178588867,\n",
       " (('source', 'Country.of.Origin'),\n",
       "  ('target', 'target_capture_kit_vendor')): 0.7403342723846436,\n",
       " (('source', 'Type'), ('target', 'project')): 0.7401650547981262,\n",
       " (('source', 'Ischemia.Time.in.Minutes'),\n",
       "  ('target', 'peripancreatic_lymph_nodes_tested')): 0.7401019334793091,\n",
       " (('source', 'MSH2_mutation'), ('target', 'loci_count')): 0.739875316619873,\n",
       " (('source', 'Ethnicity_Self_Identify'),\n",
       "  ('target', 'race')): 0.739773154258728,\n",
       " (('source', 'TP53.mutation'), ('target', 'related_files')): 0.739737868309021,\n",
       " (('source', 'Mutation.Count.ExcludingINDELs.wes'),\n",
       "  ('target', 'tumor_length_measurement')): 0.7396692037582397,\n",
       " (('source', 'GATA3.Mutation.Status'),\n",
       "  ('target', 'chip_id')): 0.7395305633544922,\n",
       " (('source', 'xCell.ImmuneScore.rna'),\n",
       "  ('target', 'contamination_error')): 0.7392734289169312,\n",
       " (('source', 'CDKN2A.mutation.status'),\n",
       "  ('target', 'chip_id')): 0.7390772700309753,\n",
       " (('source', 'Age'),\n",
       "  ('target', 'relationship_age_at_diagnosis')): 0.7386453151702881,\n",
       " (('source', 'tumor_necrosis'),\n",
       "  ('target', 'non_nodal_tumor_deposits')): 0.7385339736938477,\n",
       " (('source', 'PTEN.mutation'), ('target', 'type')): 0.7380626201629639,\n",
       " (('source', 'Ischemia.Time.in.Minutes'),\n",
       "  ('target', 'tumor_largest_dimension_diameter')): 0.7379839420318604,\n",
       " (('source', 'NOTCH1.mutation'), ('target', 'plate_name')): 0.7379834055900574,\n",
       " (('source', 'case_id'), ('target', 'code')): 0.7379173636436462,\n",
       " (('source', 'TP53.mutation.status'),\n",
       "  ('target', 'chip_id')): 0.7378031015396118,\n",
       " (('source', 'tumor_reoccur_after_treatment'),\n",
       "  ('target', 'chip_id')): 0.7378031015396118,\n",
       " (('source', 'NMF.subtype'),\n",
       "  ('target', 'aggregated_somatic_mutations')): 0.7374277114868164,\n",
       " (('source', 'BRAF_mutation'), ('target', 'loci_count')): 0.7365520000457764,\n",
       " (('source', 'TSNet.Purity.rna'),\n",
       "  ('target', 'age_at_diagnosis')): 0.7365310192108154,\n",
       " (('source', 'Label-free Proteome'),\n",
       "  ('target', 'loci_count')): 0.7361466288566589,\n",
       " (('source', 'pathologic_staging_regional_lymph_nodes_pn'),\n",
       "  ('target', 'portion_number')): 0.7356805801391602,\n",
       " (('source', 'num_smoke_per_day'),\n",
       "  ('target', 'prostatic_chips_total_count')): 0.7355826497077942,\n",
       " (('source', 'height'),\n",
       "  ('target', 'size_extraocular_nodule')): 0.7355116605758667,\n",
       " (('source', 'TMT.Plex'),\n",
       "  ('target', 'tobacco_smoking_status')): 0.7349375486373901,\n",
       " (('source', 'Tumor_Focality'), ('target', 'namespace')): 0.7344001531600952,\n",
       " ...}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isresmat.eval_to_match()\n",
    "isresmat.get_matches(top_k=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fa9a811-9250-47e3-b469-148fb9dc26b6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "my_env",
   "language": "python",
   "name": "my_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
